---
title: "input"
output: github_document
always_allow_html: true
---

```{r setup, include=FALSE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(knitr)
library(rmarkdown)
library(usmap)
library(data.table)
```

### Samples


focus on mainland foxes from Sacks only (remove one bad sample) *n=41 remaining*

figure out number of non-hybrid samples  *n=26 (14 east + 12 west)*

```{r Samples, warning=FALSE, message=FALSE, echo=FALSE, fig.height=3, fig.width = 7}

usa <- map_data("state")

pop <- read_tsv("inputstats_files/mainland_WGSmetadata.txt") %>% 
  dplyr::rename("long"=Lon,"lat"=Lat) %>% 
  mutate(`Samples (n=41)`=case_when(Deme=="hybrid" ~ "hybrid (15)",
                                    Region=="West" ~ "west (12)",
                                    TRUE ~ "east (14)")) %>% 
  mutate(depth=case_when(`Depth(x)`>18 ~ 8,
                         TRUE ~ 4)) %>% 
  filter(Dataset=="Sacks")


ggplot() +
  geom_polygon(data = usa, aes(x=long, y = lat, group = group), 
               fill = "gray90", color="gray40", linewidth=0.3) + 
  geom_point(data = pop, aes(x=long, y = lat, fill = `Samples (n=41)`),shape = 21, size=2) +
  coord_fixed(xlim = c(-123, -75),  ylim = c(26, 41.8)) +
  scale_fill_manual(values = c("dodgerblue2", "goldenrod", "red2"))+
  xlab("") + ylab("") +
  theme(panel.grid.minor=element_blank(),
        panel.grid.major = element_blank(),
        legend.position = "top",
        legend.title = element_text(size=9),
        panel.background=element_rect(color="white",fill="white",linewidth = 0.5),
        axis.line=element_line(color="black"),
        axis.text=element_text(color="black"),
        axis.ticks=element_line(color="black")) 


pop  %>% filter(Deme=="pure") %>%  group_by(`Samples (n=41)`, Sex) %>% count()
```

Split data in half:

Samples for smc++ (higher depth)

|Region |SRA	        |Sex|State|	Lat.  |  Long.  | Depth |
| ----- | ----------- | - | --- | ----- | ------- | ----- | 
|	East  |	SRR24465304	|	M	|	SC	| 33.346|	-81.743	|	26.2  | 
| East	| SRR24465268	|	F	|	AR	| 35.708|	-92.519	|	7.2   |
|	East	| SRR24465307	|	M	|	AR	| 34.287|	-91.338	|	6.4   |
|	East	| SRR24465305	|	M	|	KY	| 37.322|	-84.928	|	6.3   |
|	East	| SRR24465269	|	M	|	AR	| 33.467|	-92.169	|	6.2   |
|	East	| SRR24465309	|	F	|	MO	| 37.152|	-91.391	|	6.2   |
|	West	| SRR24465290	|	M	|	TX	| 29.508|	-103.328|	23.6  |
|	West	| SRR24465296	|	M	|	CA	| 38.490|	-122.152|	7.7   |
|	West	| SRR24465294	|	M	|	TX	| 29.924|	-101.992|	6.4   |
|	West	| SRR24465289	|	M	|	TX	| 30.213|	-102.430|	6.1   |
|	West	| SRR24465291	|	F	|	TX	| 30.684|	-101.308|	6.1   |
|	West	| SRR24465287	|	M	|	NV	| 38.821|	-116.494|	5.8   |


Samples for empirical analysis (lower depth)

|Region |SRA	        |Sex|State|	Lat.  |  Long.  | Depth |
| ----- | ----------- | - | --- | ----- | ------- | ----- | 
| East  |	SRR24465272 | M | TN | 35.543 | -88.856 |  5.7  | 
| East  |	SRR24465270 | F | AR | 36.052 | -90.973 |  5.6  | 
| East  |	SRR24465308 | F | MS | 33.937 | -89.337 |  5.1  | 
| East  |	SRR24465306 | F | AR | 33.588 | -93.391 |  4.9  | 
| East  |	SRR24465297 | F | MS | 32.766 | -90.388 |  4.8  | 
| East  |	SRR24465271 | M | AL | 32.248 | -87.791 |  4.8  | 
| West  |	SRR24465292 | M | TX | 29.561 | -104.372|  5.5  | 
| West  |	SRR24465283 | F | NM | 33.156 | -104.321|  5.5  | 
| West  |	SRR24465293 | M | TX | 30.625 | -104.067|  5.5  | 
| West  |	SRR24465288 | F | NV | 37.459 | -114.460|  5.0  | 
| West  |	SRR24465285 | M | NM | 34.956 | -103.157|  5.0  | 
| West  |	SRR24465284 | M | NM | 32.711 | -107.845|  4.9  | 


### Genic Regions

create bed file of genes ±1kb from annotation

<details>
<summary>Show code</summary>
<br>
```{r annotation, eval=FALSE, warning=FALSE,message=FALSE}

# annotation file
galba <- fread("~/Downloads/galba.gtf.gz")

# chromosome name conversion file 
chroms <- read_tsv("grayfox_renameChroms_number.txt", col_names = c("scaf","chrom")) %>% 
  separate(scaf, remove=F, c(NA,NA,NA,NA,NA,"length"))

# write file with chromosome sizes
chroms %>% select(scaf,length) %>% write_tsv("chromSizes.txt", col_names = F)

# filter annotation for genic regions
genes <- galba %>% filter(V3=="gene") %>% select(V1,V4,V5)

# add chromosome names to annotation file and remove stuff not mapping to scaffs 1-32
genescaf <- genes %>% left_join(chroms,by=c("V1"="scaf")) %>% 
  select(chrom,V4,V5, length)  %>%  na.omit()

# add 1kb to either end of each gene, 
## make sure values don't drop below zero or exceed the length of the chromosome
gene1kb <- genescaf %>% mutate(start=V4-1000, end=V5+1000) %>% 
  mutate(newstart=case_when(start<0 ~ 1, TRUE ~ start)) %>% 
  mutate(newend=case_when(end>as.numeric(length) ~ as.numeric(length), TRUE ~ end))

gene1kb %>% select(chrom,newstart,newend) %>% write_tsv("genes1kb.bed", col_names = F)

```
</details>


<br>

remove genic regions and compute summary statistics on half of data

<details>
<summary>Show code</summary>
<br>
```{bash genic, eval=FALSE}

module load bedtools2
module load bcftools

bedtools sort -i genes1kb.bed > genicregions.1kb.bed

#removes genic regions
bcftools view -T ^genicregions.1kb.bed grayfox_filtered.renameChroms.Mainland.ACgr61_DPgr165lt500.vcf.gz -Oz -o grayfox_mainland_nogenes.vcf.gz 

bcftools index -t grayfox_mainland_nogenes.vcf.gz

#split by chromosome
bcftools index -s grayfox_mainland_nogenes.vcf.gz | cut -f 1 | while read C; do bcftools view -O z -o split.${C}.vcf.gz grayfox_mainland_nogenes.vcf.gz "${C}" ; done

```
</details>

### Compute Summary Statistics

calculate pi, segregating sites, heterozygosity per site on n=12 empirical samples

<details>
<summary>Show code</summary>
<br>
```{bash vcftools-stats, eval=FALSE}

#!/bin/sh
#SBATCH --job-name=statsvcfsite
#SBATCH --output=/scratch1/marjanak/statsvcfsite.out
#SBATCH --error=/scratch1/marjanak/statsvcfsite.err
#SBATCH --partition=qcb
#SBATCH --time=30:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem-per-cpu=8000MB
#SBATCH --mail-type=END,FAIL # notifications for job done & fail
#SBATCH --mail-user=marjanak@usc.edu

module load vcftools

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Invariant/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.ACgr61_DPgr165lt500.gvcf.gz --keep east6.txt --exclude-bed genicregions.1kb.bed --site-pi --out east6_pi_site.out

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Invariant/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.ACgr61_DPgr165lt500.gvcf.gz --keep west6.txt --exclude-bed genicregions.1kb.bed --site-pi --out west6_pi_site.out


vcftools --gzvcf /project/jazlynmo_738/Maria/grayfox_mainland_nogenes.vcf.gz --keep east6.txt --snpdensity 10 --out east6_S_10bp.out

vcftools --gzvcf /project/jazlynmo_738/Maria/grayfox_mainland_nogenes.vcf.gz --keep west6.txt --snpdensity 10 --out west6_S_10bp.out



```
</details>

<br>

SNP Density

<details>
<summary>Show code</summary>
<br>
```{bash vcfsnpden, eval=FALSE}

#!/bin/sh
#SBATCH --job-name=bcfsnpden
#SBATCH --output=/scratch1/marjanak/bcfsnpden.out
#SBATCH --error=/scratch1/marjanak/bcfsnpden.err
#SBATCH --partition=qcb
#SBATCH --time=30:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem-per-cpu=8000MB
#SBATCH --mail-type=END,FAIL # notifications for job done & fail
#SBATCH --mail-user=marjanak@usc.edu

module load bcftools

bcftools stats -s SRR24465270,SRR24465271,SRR24465272,SRR24465297,SRR24465306,SRR24465308 -T ^genicregions.1kb.bed /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.ACgr61_DPgr165lt500.vcf.gz 


bcftools stats -s SRR24465283,SRR24465284,SRR24465285,SRR24465288,SRR24465292,SRR24465293 -T ^genicregions.1kb.bed /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.ACgr61_DPgr165lt500.vcf.gz 


bcftools stats -s SRR24465270,SRR24465271,SRR24465272,SRR24465297,SRR24465306,SRR24465308 -T ^genicregions.1kb.bed /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/Canfam3.1_filtered.renameChroms.Mainland.ACgr61_DPgr165lt500.vcf.gz


bcftools stats -s SRR24465283,SRR24465284,SRR24465285,SRR24465288,SRR24465292,SRR24465293 -T ^genicregions.1kb.bed /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/Canfam3.1_filtered.renameChroms.Mainland.ACgr61_DPgr165lt500.vcf.gz



```
</details>


```{bash scratch, echo=FALSE, eval=FALSE}


#!/bin/sh
#SBATCH --job-name=sitesvcf
#SBATCH --output=/scratch1/marjanak/sitesvcf.out
#SBATCH --error=/scratch1/marjanak/sitesvcf.err
#SBATCH --partition=qcb
#SBATCH --time=24:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem-per-cpu=8000MB
#SBATCH --mail-type=END,FAIL # notifications for job done & fail
#SBATCH --mail-user=marjanak@usc.edu

module load vcftools

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Invariant/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.ACgr61_DPgr165lt500.gvcf.gz --keep east6.txt --exclude-bed genicregions.1kb.bed --TajimaD 10000 --out east6_tajD_10kb.out

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Invariant/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.ACgr61_DPgr165lt500.gvcf.gz --keep west6.txt --exclude-bed genicregions.1kb.bed --TajimaD 10000 --out west6_tajD_10kb.out
```


<br>

### Mapped to Gray Fox Genome

Average nucleotide diversity 1.8x higher in the east (n=6) compared to the west (n=6) when mapped to grayfox genome


```{r pi, warning=FALSE, message=FALSE, echo=FALSE, fig.height=4, fig.width = 10}
eastpi <- read_tsv("east6_pi_10kb.out.windowed.pi") %>%
  mutate(chr_position = ((BIN_START + BIN_END)/2)/1000000) %>%
  mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) 

westpi <- read_tsv("west6_pi_10kb.out.windowed.pi") %>%
  mutate(chr_position = ((BIN_START + BIN_END)/2)/1000000) %>%
  mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) 

allpi <- bind_rows("east"=eastpi,"west"=westpi,.id = "pop") %>%
  mutate(avg=case_when(pop=="east" ~ 0.000108,
                       TRUE ~ 0.0000611))

allpi %>% group_by(pop) %>% summarise(avgpi=mean(PI), medpi=median(PI), sdpi=sd(PI))

ggplot() + 
  geom_line(data=allpi, aes(x=chr_position, y=PI, color=pop), size=0.15) +
  scale_alpha_manual(values = rep(c(1, 0.7), 100)) +
  scale_color_manual(values = c("#4d9aff", "#f2675f"))+
  xlab("Position (Mbp)") +
  facet_grid(cols = vars(CHROM), rows=vars(pop),
             space = "free_x",
             scales = "free_x") +     
  ylab(expression(pi))+
  #scale_y_continuous(limits = c(0,0.025), breaks = c(0, 0.01, 0.02))+
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text = element_text(colour = "black", size = 12),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 14),
    legend.position="top",
    legend.title = element_blank(),
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 14),   
    panel.spacing = unit(0.0, "cm"),
    strip.background=element_rect(fill="gray90"),
    axis.text.x = element_blank())  + 
    geom_hline(data = allpi, aes(yintercept = avg), linetype="dashed")
```


### Mapped to Canfam3.1


Average nucleotide diversity 2.2x higher in the west (n=6) compared to the east (n=6) when mapped to canfam3.1


```{r pi-cf, warning=FALSE, message=FALSE, echo=FALSE, fig.height=4, fig.width = 11}
cfeastpi <- read_tsv("cf_east6_pi_10kb.out.windowed.pi") %>%
  mutate(chr_position = ((BIN_START + BIN_END)/2)/1000000) %>%
  mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) 

cfwestpi <- read_tsv("cf_west6_pi_10kb.out.windowed.pi") %>%
  mutate(chr_position = ((BIN_START + BIN_END)/2)/1000000) %>%
  mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) 

cfallpi <- bind_rows("east"=cfeastpi,"west"=cfwestpi,.id = "pop") %>%
  mutate(avg=case_when(pop=="east" ~ 0.0000800,
                       TRUE ~ 0.000173))

cfallpi %>% group_by(pop) %>% summarise(avgpi=mean(PI), medpi=median(PI), sdpi=sd(PI))

ggplot() + 
  geom_line(data=cfallpi, aes(x=chr_position, y=PI, color=pop), size=0.15) +
  scale_alpha_manual(values = rep(c(1, 0.7), 100)) +
  scale_color_manual(values = c("#4d9aff", "#f2675f"))+
  xlab("Position (Mbp)") +
  facet_grid(cols = vars(CHROM), rows=vars(pop),
             space = "free_x",
             scales = "free_x") +     
  ylab(expression(pi))+
  #scale_y_continuous(limits = c(0,0.025), breaks = c(0, 0.01, 0.02))+
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text = element_text(colour = "black", size = 12),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 14),
    legend.position="top",
    legend.title = element_blank(),
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 14),   
    panel.spacing = unit(0.0, "cm"),
    strip.background=element_rect(fill="gray90"),
    axis.text.x = element_blank()) + 
  geom_hline(data = cfallpi, aes(yintercept = avg), linetype="dashed")
```



<details>
<summary>Show code</summary>
<br>
```{bash vcfhet, eval=FALSE}

#!/bin/sh
#SBATCH --job-name=vcfhet
#SBATCH --output=/scratch1/marjanak/vcfhet.out
#SBATCH --error=/scratch1/marjanak/vcfhet.err
#SBATCH --partition=qcb
#SBATCH --time=20:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem-per-cpu=8000MB
#SBATCH --mail-type=END,FAIL # notifications for job done & fail
#SBATCH --mail-user=marjanak@usc.edu

module load vcftools

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.ACgr61_DPgr165lt500.vcf.gz --keep empirical6.txt --exclude-bed genicregions.1kb.bed --het --out grayfox.het

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/Canfam3.1_filtered.renameChroms.Mainland.ACgr61_DPgr165lt500.vcf.gz --keep empirical6.txt --exclude-bed genicregions.1kb.bed --het --out canfam3.1.het


```
</details>

Average heterozygosity (number of observed heterozygous sites divided by total number of variant sites) across samples in the empirical dataset (n=12)

```{r het, warning=FALSE, message=FALSE, echo=FALSE, fig.height=5, fig.width = 6}

het <- read_tsv("het.het") 

hettab <- het %>% group_by(GENOME,POP) %>% summarise(avghet=mean(Het), medhet=median(Het), sdhet=sd(Het))

hettab

ggplot(hettab,aes(x=interaction(POP,GENOME), y=avghet, fill=POP)) +
  geom_col()+ 
  geom_errorbar(aes(ymin=avghet-sdhet, ymax=avghet+sdhet), width=0.2) +
  scale_fill_manual(values = c("#4d9aff", "#f2675f"))+
  theme_bw()+
  ylab("Heterozygosity ± SD")+
  theme(axis.title.x = element_blank(),
        axis.text=element_text(color="black",size=10),
        legend.position = "none")

```


Segregating sites

~15M SNPs when mapped to Canfam3.1 vs ~350K SNPs when mapped to Grayfox

pop values are averaged across n=6 individuals

the majority of SNPs are homozygous for the non-reference allele 

number of homozygous reference and heterozygous alleles higher in the west when mapped to Camfam3.1 but higher in the east when mapped to Grayfox

Grayfox genome from an eastern individual, hence lowest number of non-reference alleles in the east


```{r segsites, warning=FALSE, message=FALSE, echo=FALSE, fig.height=7, fig.width = 6}

nvargf <- allpi %>% group_by(pop) %>% summarise(var=sum(N_VARIANTS))

nvarcf <- cfallpi %>% group_by(pop) %>% summarise(var=sum(N_VARIANTS))

nvar <- bind_rows("gf"=nvargf,"cf"=nvarcf,.id = "genome")


snps <- read_tsv("snps.txt")

snpgrp <- snps %>% group_by(pop,genome) %>% 
  summarise(avgnRefHom=mean(nRefHom),avgnNonRefHom=mean(nNonRefHom),avgnHets=mean(nHets)) %>% 
  pivot_longer(cols = starts_with("avg"), names_to = "SNPs", values_to="count") %>% mutate(count=as.integer(count))


ggplot(snpgrp,aes(x=SNPs, y=count, fill=pop)) +
  geom_col(position="dodge") + 
  geom_text(data=subset(snpgrp,genome=="canfam31"), aes(y=-1000000,label = count),size=3.5,position = position_dodge(width = 1))+ 
    geom_text(data=subset(snpgrp,genome=="grayfox"), aes(y=-20000,label = count),size=3.5,position = position_dodge(width = 1))+ 
  scale_fill_manual(values = c("#4d9aff", "#f2675f"))+
  theme_bw()+
  ylab("Variants")+
  theme(axis.title.x = element_blank(),
        axis.text=element_text(color="black",size=10),
        legend.position = "none")+
  facet_wrap(~genome, scales="free",ncol=1)

library(scales)
library(janitor)

snpper <- snpgrp %>% 
  group_by(pop,genome) %>% 
  mutate(prop = count/sum(count)) %>%  
  mutate(labels = scales::percent(prop, accuracy=0.01)) %>% 
  mutate(new=paste(count,labels,sep=" "))

counts <- snpper %>% select(pop,genome,SNPs,count) %>% 
  pivot_wider(names_from=SNPs,values_from=c(count)) %>% 
  arrange(genome) %>% adorn_totals(where="col") %>% mutate_if(is.integer, as.character)


perc <- snpper %>% select(pop,genome,SNPs,labels) %>% 
  pivot_wider(names_from=SNPs,values_from=c(labels)) %>% 
  arrange(genome) 


snpcounts <- bind_rows(counts,perc) %>% arrange(genome,pop) %>%
  mutate(across(everything(), ~ replace(.x, is.na(.x), "")))


snpcounts


```
