---
title: "Mainland"
output: 
  github_document:
    toc: true
always_allow_html: true
---

```{r setup, include=FALSE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(knitr)
library(rmarkdown)
library(usmap)
library(data.table)
library(scales)
library(janitor)
library(kableExtra)
library(ggpubr)
```



## Samples 

Whole genome sequencing of gray fox 

Data from [ncbi.nlm.nih.gov/bioproject/PRJNA966176/](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA966176/)

>Preckler-Quisquater et al. (2023). Can demographic histories explain long-term isolation and recent pulses of asymmetric gene flow between highly divergent grey fox lineages? Molecular Ecology, 32, 5323â€“5337. https://doi.org/10.1111/mec.17105

We analyzed n=26 samples (n=14 from East and n=12 from West)

```{r Samples, echo=FALSE, fig.height=2.5, message=FALSE, warning=FALSE}

usa <- map_data("state")

pop <- read_tsv("Mainland_files/input/mainland_WGSmetadata.txt") %>% 
  dplyr::rename("long"=Lon,"lat"=Lat) %>% 
  filter(Dataset=="Sacks") %>% 
  filter(Deme=="pure") %>% 
  mutate(alf= case_when(`Depth(x)`<4.5 ~ "drop", TRUE ~ "keep"))


ggplot() +
  geom_polygon(data = usa, aes(x=long, y = lat, group = group), 
               fill = "white", color="gray70", linewidth=0.5) + 
  geom_point(data=pop, aes(x=long, y = lat, fill=Region, color=alf),
             size=2,shape=21,stroke=0.5) +
  coord_fixed(xlim = c(-123, -75),  ylim = c(26, 41.8)) +
  scale_fill_manual(values = c("goldenrod2", "aquamarine3"))+
  scale_color_manual(values = c("red","black"))+
  xlab("") + ylab("") +
  theme(panel.grid.minor=element_blank(),
        panel.grid.major = element_blank(),
        legend.position = c(0.2,0.1),
        legend.key = element_rect(colour = NA, fill = NA),
        legend.spacing.y = unit(-0.1, "cm"),
        legend.title = element_blank(),
        panel.background=element_rect(color="white",fill="white",linewidth = 0.5),
        axis.line=element_blank(),
        axis.text=element_blank(),
        axis.ticks=element_blank()) +
  guides(colour = "none",
         fill = guide_legend(nrow = 1))


```

Eastern samples outlined in red were dropped (sequencing depth <4.5x) for analyses sensitive to unequal sample sizes (e.g. site frequency spectra)


## Reference genome

>Armstrong et al. (2024). Chromosome-level assembly of the gray fox (Urocyon cinereoargenteus) confirms the basal loss of PRDM9 in Canidae, G3 Genes|Genomes|Genetics, 14:4jkae034, https://doi.org/10.1093/g3journal/jkae034

<img src="Mainland_files/input/canid_synteny.png" width="400"/>

## Demographies

SMC++ Output

```{r Demography, echo=FALSE, fig.height=4, message=FALSE, warning=FALSE}
#Grayfox West
gfplotwest <- read_csv("Mainland_files/input/gf_west_plot.csv")
#Grayfox East
gfploteast <- read_csv("Mainland_files/input/gf_east_plot.csv")
#Combine Grayfox
gfplot<- bind_rows(gfplotwest,gfploteast)

#Canfam4 West
cf4plotwest <- read_csv("Mainland_files/input/cf4_west_plot.csv")
#Canfam4 East
cf4ploteast <- read_csv("Mainland_files/input/cf4_east_plot.csv")
#Combine Canfam4
cf4plot<- bind_rows(cf4plotwest,cf4ploteast) 

#Canfam3 West
cf3plotwest <- read_csv("Mainland_files/input/cf3_west_plot.csv")
#Canfam3 East
cf3ploteast <- read_csv("Mainland_files/input/cf3_east_plot.csv")
#Combine Canfam3
cf3plot<- bind_rows(cf3plotwest,cf3ploteast)


#Combine Grayfox and Canfam4
cfgf <- bind_rows("Grayfox"=gfplot, "Canfam3"=cf3plot, "Canfam4"=cf4plot, .id="genome") %>% 
  mutate(pop=case_when(label=="WES" ~ "West", TRUE ~ "East")) %>% 
  filter(x>1e3& x<1e6) %>% select(genome,pop,x,y)


cx_breaks = c(0,10,100,1e3,1e4,1e5,1e6)
cy_breaks = c(0,1e3,2e3,3e3,5e3,8e3,125e2,2e4,3e4,45e3,7e4,1e5,2e5,5e5)


ggplot()+
  geom_line(data=cfgf,aes(x=x,y=y,color=pop,linetype=genome),linewidth=1) + 
  scale_y_continuous(trans = "log",breaks=cy_breaks,labels = ~ format(.x, scientific = FALSE))+
  scale_x_continuous(trans = "log",breaks=cx_breaks,labels = ~ format(.x, scientific = FALSE))+ 
  scale_linetype_manual(values=c("dotdash","solid","dotted"))+
  labs(x ="Years", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11),
        legend.position = "top",legend.title = element_blank()) +
  scale_color_manual(values = c("goldenrod2","aquamarine3"))


pcfgf <- cfgf %>% 
  rename("Ne"="y") %>% 
  group_by(genome,pop,Ne) %>% 
  summarise(end=max(x),start=min(x)) %>%  
  arrange(genome,pop,desc(start)) %>%
  mutate(Ne=plyr::round_any(Ne,100),start=plyr::round_any(start,100),end=plyr::round_any(end,100)) %>% ungroup()


pcf4e <- pcfgf %>% filter(genome=="Canfam4" & pop=="East") %>% select(start,end,Ne) 
pcf3e <- pcfgf %>% filter(genome=="Canfam3" & pop=="East") %>% select(start,end,Ne) 
pgfe <- pcfgf %>% filter(genome=="Grayfox" & pop=="East") %>% select(start,end,Ne)  

pcf4w <- pcfgf %>% filter(genome=="Canfam4" & pop=="West") %>% select(start,end,Ne) 
pcf3w <- pcfgf %>% filter(genome=="Canfam3" & pop=="West") %>% select(start,end,Ne) 
pgfw <- pcfgf %>% filter(genome=="Grayfox" & pop=="West") %>% select(start,end,Ne) 

kable(pcf4e, "html", escape = FALSE,format.args = list(big.mark = ",") ) %>%
  kable_styling(full_width = FALSE, position = "float_left") %>% 
  add_header_above(c("East" = 3))%>% 
  add_header_above(c("Canfam4" = 3))

kable(pcf3e, "html", escape = FALSE,format.args = list(big.mark = ",") ) %>%
  kable_styling(full_width = FALSE, position = "float_left") %>% 
  add_header_above(c("East" = 3))%>% 
  add_header_above(c("Canfam3" = 3))

kable(pgfe, "html", escape = FALSE,format.args = list(big.mark = ",") ) %>%
  kable_styling(full_width = FALSE, position = "left") %>% 
  add_header_above(c("East" = 3))%>% 
  add_header_above(c("Grayfox" = 3))

```

<br>

```{r Demography2, echo=FALSE, fig.height=8, message=FALSE, warning=FALSE}

ggplot()+
  geom_line(data=cfgf,aes(x=x,y=y,color=pop),linewidth=1) + 
  scale_y_continuous(trans = "log",breaks=cy_breaks,labels = ~ format(.x, scientific = FALSE))+
  scale_x_continuous(trans = "log",breaks=cx_breaks,labels = ~ format(.x, scientific = FALSE))+ 
  labs(x ="Years", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11),
        legend.position = "top",legend.title = element_blank()) +
  scale_color_manual(values = c("goldenrod2","aquamarine3")) + 
  facet_wrap(~genome,nrow=3,scales="free_y")


   
kable(pcf4w, "html", escape = FALSE,format.args = list(big.mark = ",") ) %>%
  kable_styling(full_width = FALSE, position = "float_left") %>% 
  add_header_above(c("West" = 3))%>% 
  add_header_above(c("Canfam4" = 3))

kable(pcf3w, "html", escape = FALSE,format.args = list(big.mark = ",") ) %>%
  kable_styling(full_width = FALSE, position = "float_left") %>% 
  add_header_above(c("West" = 3))%>% 
  add_header_above(c("Canfam3" = 3))

kable(pgfw, "html", escape = FALSE,format.args = list(big.mark = ",") ) %>%
  kable_styling(full_width = FALSE, position = "left") %>% 
  add_header_above(c("West" = 3)) %>% 
  add_header_above(c("Grayfox" = 3))




#kable(list(pcf4e,pcf3e,pgfe),caption="East") %>% kable_styling(full_width = F) 
#kable(list(pcf4w,pcf3w,pgfw),caption="West") %>% kable_styling(full_width = F)
```

<br>
<br>
<br>
<br>


## Recombination rates

Pyrho output

```{r Recombination, echo=FALSE, fig.width=7, fig.height=4,message=FALSE, warning=FALSE}
syn_avg <- data.frame(
  genome=c("Canfam4","Canfam4","Canfam3","Canfam3","Grayfox","Grayfox"),
  pop=c("East","West","East","West","East","West"),
  recomb=c(4.15505e-08,6.63379e-08,5.54515e-08,1.04263e-07,1.34378e-08,2.08488e-08))

kable(syn_avg,digits = 12 )

synteny<-read_tsv("Mainland_files/input/syn_chrom_mean.txt")

ggplot(synteny,aes(x=mean.gf, y=mean.cf3,color = pop)) +
  geom_point() + 
  geom_smooth(method = "lm")+ 
  stat_cor()+
  facet_wrap(~pop,scales="free")+  
  scale_color_manual(values = c("goldenrod2", "aquamarine3"))+
  xlab("Grayfox Average Recombination Rate") +
  ylab("Canfam3 Average Recombination Rate") + 
  theme_classic() +
  theme(legend.position="top", axis.text = element_text(color="black"),
        plot.margin = margin(5, 15, 5, 5))

```

