---
title: "Filtered 2024"
output: 
  github_document:
    toc: true
---

```{r setup, include=FALSE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(knitr)
library(rmarkdown)
library(usmap)
library(data.table)
library(scales)
library(janitor)
```

## Previous Results
 
>Preckler-Quisquater et al. (2023). Can demographic histories explain long-term isolation and recent pulses of asymmetric gene flow between highly divergent grey fox lineages? Molecular Ecology, 32, 5323–5337. https://doi.org/10.1111/mec.17105

Identified significantly **lower overall heterozygosity** in the **eastern (6.0×10<sup>−4</sup>)** relative to the **western (9.8×10<sup>−4</sup>)** population (p«.001) and observed a decrease in heterozygosity along a northwest gradient in the eastern lineage:

![Spatial patterns of heterozygosity](https://onlinelibrary.wiley.com/cms/asset/95887713-c8e0-4754-a42f-c3f0aeedb70a/mec17105-fig-0006-m.jpg)

## Reanalysis

### Samples 
```{r Samples, warning=FALSE, message=FALSE, echo=FALSE, fig.height=6, fig.width = 10}

usa <- map_data("state")
world_map <- map_data("world")

east6emp <- c("SRR24465306","SRR24465269","SRR24465307","SRR24465297","SRR24465272","SRR24465305")

west6emp <- c("SRR24465296","SRR24465288","SRR24465292","SRR24465293","SRR24465285","SRR24465294")

east6dem <- c("SRR24465268","SRR24465309","SRR24465270","SRR24465308","SRR24465271","SRR24465304")

west6dem <- c("SRR24465287","SRR24465284","SRR24465283","SRR24465290","SRR24465289","SRR24465291")

empsamps <- bind_cols(as_tibble(east6emp),as_tibble(west6emp)) %>% rename("East"="value...1", "West"="value...2")

knitr::kable(empsamps, format = "html", caption = "Samples for Empirical Analysis")

demsamps <- bind_cols(as_tibble(east6dem),as_tibble(west6dem)) %>% rename("East"="value...1", "West"="value...2")


pop <- read_tsv("inputstats_files/mainland_WGSmetadata.txt") %>% 
  dplyr::rename("long"=Lon,"lat"=Lat) %>% 
  filter(Dataset=="Sacks") %>% filter(`Depth(x)`>4.5) %>% filter(Deme=="pure") %>% 
  mutate(Subset=case_when(SRA %in% east6emp ~ "empirical",
                          SRA %in% west6emp ~ "empirical",
                          TRUE ~ "smc++"))

ggplot() +
  geom_polygon(data = world_map, aes(x=long, y = lat, group = group), 
               fill = "gray95", color="gray40", linewidth=0.3) + 
  geom_polygon(data = usa, aes(x=long, y = lat, group = group), 
               fill = "gray95", color="gray70", linewidth=0.3) + 
  geom_point(data=pop, aes(x=long, y = lat, shape=Sex,color=Subset), size=3) +
  coord_fixed(xlim = c(-123, -75),  ylim = c(26, 41.8)) +
  scale_color_manual(values = c("dodgerblue2", "goldenrod"))+
  xlab("") + ylab("") +
  theme(panel.grid.minor=element_blank(),
        panel.grid.major = element_blank(),
        legend.position = "top",
        legend.title = element_text(size=9),
        panel.background=element_rect(color="white",fill="white",linewidth = 0.5),
        axis.line=element_line(color="black"),
        axis.text=element_text(color="black"),
        axis.ticks=element_line(color="black")) 
```

### Genic Regions

create bed file of genes ±1kb from annotation

<details>
<summary>Show code</summary>
<br>
```{r annotation, eval=FALSE, warning=FALSE,message=FALSE}


##Canfam3.1 annotation

chroms <- read_tsv("sumstats_files/canfamchrom.txt")

cfgenes <- read_tsv("sumstats_files/canFam3.1_ensembleGenes", col_names = T) %>% 
  left_join(chroms, by=c("chrom"="Chromosome")) %>% 
  filter(str_detect(chrom, "chr\\d")) %>%  
  mutate(newstart=txStart-1000, newend=txEnd+1000) %>% 
  mutate(start=case_when(newstart<0 ~ 1, TRUE ~ newstart)) %>%
  mutate(end=case_when(newend>as.numeric(Size) ~ as.numeric(Size), TRUE ~ newend)) %>%
  select(chrom,start,end)

mergecf <- bed_merge(cfgenes)

mergecf %>% write_tsv("sumstats_files/cfgenes1kb.bed", col_names = F)


#Grayfox liftover annotation

galba <- fread("sumstats_files/galba.gtf.gz")

chroms <- read_tsv("inputstats_files/grayfox_renameChroms_number.txt", col_names = c("scaf","chrom")) %>% 
  separate(scaf, remove=F, c(NA,NA,NA,NA,NA,"length"))

gfgenes <- galba %>% left_join(chroms,by=c("V1"="scaf")) %>% na.omit() %>%
  mutate(newstart=V4-1000, newend=V5+1000) %>% 
  mutate(start=case_when(newstart<0 ~ 1, TRUE ~ newstart)) %>%
  mutate(end=case_when(newend>as.numeric(length) ~ as.numeric(length), TRUE ~ newend)) %>%
  select(chrom,start,end)

mergegf <- bed_merge(gfgenes)

mergegf %>% write_tsv("sumstats_files/gfgenes1kb.bed", col_names = F)

```
</details>





### Summary Statistics 
<details>
<summary>Show code</summary>
<br>
```{bash varpi10kb, eval=FALSE}
#!/bin/sh
#SBATCH --job-name=varpi10kb
#SBATCH --output=/scratch1/marjanak/varpi10kb.out
#SBATCH --error=/scratch1/marjanak/varpi10kb.err
#SBATCH --partition=qcb
#SBATCH --time=30:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem-per-cpu=8000MB
#SBATCH --mail-type=END,FAIL # notifications for job done & fail
#SBATCH --mail-user=marjanak@usc.edu

module load vcftools
module load bcftools

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz --keep east6.txt --exclude-bed gfgenes1kb.bed --window-pi 10000 --out gf_east6_pi_10kb.out

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/Canfam3.1_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz --keep east6.txt --exclude-bed cfgenes1kb.bed --window-pi 10000 --out cf_east6_pi_10kb.out

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz --keep west6.txt --exclude-bed gfgenes1kb.bed --window-pi 10000 --out gf_west6_pi_10kb.out

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/Canfam3.1_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz --keep west6.txt --exclude-bed cfgenes1kb.bed --window-pi 10000 --out cf_west6_pi_10kb.out

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz --keep empirical.txt --exclude-bed gfgenes1kb.bed --het --out grayfox.het

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/Canfam3.1_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz --keep empirical.txt --exclude-bed cfgenes1kb.bed --het --out canfam31.het

bcftools stats -S empirical.txt -T ^gfgenes1kb.bed /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz > grayfox.seg

bcftools stats -S empirical.txt -T ^cfgenes1kb.bed /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/Canfam3.1_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz > canfam31.seg



```
</details>


### Split VCF by pop and count alleles 
<details>
<summary>Show code for Grayfox</summary>
<br>
```{bash sepvcf, eval=FALSE}
#!/bin/sh
#SBATCH --job-name=sepvcf
#SBATCH --output=/scratch1/marjanak/sepvcf.out
#SBATCH --error=/scratch1/marjanak/sepvcf.err
#SBATCH --partition=qcb
#SBATCH --time=10:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem-per-cpu=8000MB
#SBATCH --mail-type=END,FAIL # notifications for job done & fail
#SBATCH --mail-user=marjanak@usc.edu

module load bcftools

bcftools view -Oz -S west6.txt /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz > west6.vcf.gz

bcftools view -Oz -S east6.txt /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz > east6.vcf.gz

bcftools index -t west6.vcf.gz

bcftools index -t east6.vcf.gz

vcftools --gzvcf east6.vcf.gz --counts --exclude-bed gfgenes1kb.bed

vcftools --gzvcf west6.vcf.gz --counts --exclude-bed gfgenes1kb.bed
```
</details>



<details>
<summary>Show code for CanFam3.1</summary>
<br>
```{bash cfsepvcf, eval=FALSE}
#!/bin/sh
#SBATCH --job-name=cfsepvcf
#SBATCH --output=/scratch1/marjanak/cfsepvcf.out
#SBATCH --error=/scratch1/marjanak/cfsepvcf.err
#SBATCH --partition=qcb
#SBATCH --time=10:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem-per-cpu=8000MB
#SBATCH --mail-type=END,FAIL # notifications for job done & fail
#SBATCH --mail-user=marjanak@usc.edu

module load bcftools
module load vcftools

bcftools view -Oz -S west6.txt /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/Canfam3.1_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz > cfwest6.vcf.gz

bcftools view -Oz -S east6.txt /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/Canfam3.1_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz > cfeast6.vcf.gz

bcftools index -t cfwest6.vcf.gz

bcftools index -t cfeast6.vcf.gz

vcftools --gzvcf cfeast6.vcf.gz --counts --exclude-bed cfgenes1kb.bed --out cfeast

vcftools --gzvcf cfwest6.vcf.gz --counts --exclude-bed cfgenes1kb.bed --out cfwest


```
</details>




```{r gfcounts, warning=FALSE, eval=FALSE, message=FALSE, echo=FALSE}
eastcount <- read_tsv("sumstats_files/east.frq.count") %>% 
  separate(`{ALLELE:COUNT}`,c("geno1","count1","geno2","count2")) %>%
  mutate(category=case_when(count1==N_CHR ~ "fixed anc",
                            count2==N_CHR ~ "fixed der",
                            TRUE ~ "polymorphic"))

westcount <- read_tsv("sumstats_files/west.frq.count") %>% 
  separate(`{ALLELE:COUNT}`,c("geno1","count1","geno2","count2")) %>%
  mutate(category=case_when(count1==N_CHR ~ "fixed anc",
                            count2==N_CHR ~ "fixed der",
                            TRUE ~ "polymorphic"))

westab <- westcount %>% group_by(category) %>% count() %>% 
  pivot_wider(names_from = category, values_from=n)

eastab <- eastcount %>% group_by(category) %>% count() %>% 
  pivot_wider(names_from = category, values_from=n)

botab <- bind_rows("east"=eastab,"west"=westab,.id = "pop")

joincount <- full_join(eastcount,westcount,by=c("CHROM","POS","N_ALLELES","geno1","geno2"), suffix=c(".east",".west") )

alltab <- joincount %>% group_by(category.east,category.west) %>% 
  dplyr::summarise(count=n())  %>% ungroup() %>% arrange(desc(count)) %>% 
  mutate(perc = percent(count / sum(count), accuracy = 0.1)) %>% 
  rename("east"=category.east, "west"=category.west)


## canfam3.1

cfeastcount <- fread("sumstats_files/cfeast.frq.count",col.names = c("CHROM","POS","N_ALLELES","N_CHR","allele1","count1","allele2","count2"))

cfwestcount <- fread("sumstats_files/cfwest.frq.count",col.names = c("CHROM","POS","N_ALLELES","N_CHR","allele1","count1","allele2","count2"))

cfjoincount <- full_join(cfeastcount,cfwestcount,by=c("CHROM","POS"), suffix=c(".east",".west") )

cfeastct <- cfjoincount %>% 
  mutate(east=case_when(count1.east==N_CHR.east ~ "fixed anc",
                        count2.east==N_CHR.east ~ "fixed der",
                        TRUE ~ "polymorphic"))  %>% 
  mutate(west=case_when(count1.west==N_CHR.west ~ "fixed anc",
                        count2.west==N_CHR.west ~ "fixed der",
                        TRUE ~ "polymorphic"))

cftab <- cfeastct %>% group_by(east,west) %>% 
  dplyr::summarise(count=n())  %>% ungroup() %>% arrange(desc(count)) %>% 
  mutate(perc = percent(count / sum(count), accuracy = 0.1))


jointab <- inner_join(alltab,cftab,by=c("east","west"), suffix=c(".gf",".cf") ) %>% adorn_totals()

write_tsv(jointab,"sumstats_files/jointab.tsv")
```


```{r tablecount, warning=FALSE, eval=TRUE, message=FALSE, echo=FALSE}

jointab <- read_tsv("sumstats_files/jointab.tsv")



knitr::kable(jointab, format = "html")

```