---
title: "Filtered 2024"
output: 
  github_document:
    toc: true
---

```{r setup, include=FALSE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(knitr)
library(rmarkdown)
library(usmap)
library(data.table)
library(scales)
library(janitor)
```

## Previous Results
 
>Preckler-Quisquater et al. (2023). Can demographic histories explain long-term isolation and recent pulses of asymmetric gene flow between highly divergent grey fox lineages? Molecular Ecology, 32, 5323–5337. https://doi.org/10.1111/mec.17105

Identified significantly **lower overall heterozygosity** in the **east (6.0×10<sup>−4</sup>)** relative to the **west (9.8×10<sup>−4</sup>)** (p«.001) and observed a decrease in heterozygosity along a northwest gradient in the eastern lineage:

## Reanalysis

### Samples 
```{r Samples, echo=FALSE, fig.height=3, message=FALSE, warning=FALSE}
usa <- map_data("state")
world_map <- map_data("world")

east6emp <- c("SRR24465306","SRR24465269","SRR24465307","SRR24465297","SRR24465272","SRR24465305")

west6emp <- c("SRR24465296","SRR24465288","SRR24465292","SRR24465293","SRR24465285","SRR24465294")

east6dem <- c("SRR24465268","SRR24465309","SRR24465270","SRR24465308","SRR24465271","SRR24465304")

west6dem <- c("SRR24465287","SRR24465284","SRR24465283","SRR24465290","SRR24465289","SRR24465291")

empsamps <- bind_cols(as_tibble(east6emp),as_tibble(west6emp)) %>% rename("East"="value...1", "West"="value...2")

knitr::kable(empsamps, format = "html", caption = "Samples for Empirical Analysis")

demsamps <- bind_cols(as_tibble(east6dem),as_tibble(west6dem)) %>% rename("East"="value...1", "West"="value...2")


HETS <- c("SRR24465297","SRR24465269","SRR24465270","SRR24465308","SRR24465275","SRR24465286","SRR24465305","SRR24465272")


pop <- read_tsv("inputstats_files/mainland_WGSmetadata.txt") %>% 
  dplyr::rename("long"=Lon,"lat"=Lat) %>% 
  filter(Dataset=="Sacks") %>% 
  filter(`Depth(x)`>4.5) %>% 
  mutate(depth=round(`Depth(x)`,0)) %>% mutate(depth=paste0(depth,"×")) %>% 
  filter(Deme=="pure")   %>% 
  mutate(Subset=case_when(SRA %in% east6emp ~ "empirical",
                          SRA %in% west6emp ~ "empirical",
                          TRUE ~ "smc++"))  %>% 
  mutate(hets=case_when(SRA %in% HETS ~ "HET",
                        TRUE ~ "NO"))
 

ggplot() +
  geom_polygon(data = world_map, aes(x=long, y = lat, group = group), 
               fill = "gray95", color="gray40", linewidth=0.3) + 
  geom_polygon(data = usa, aes(x=long, y = lat, group = group), 
               fill = "gray95", color="gray70", linewidth=0.3) + 
  geom_point(data=pop, aes(x=long, y = lat,color=hets), size=3) +
  coord_fixed(xlim = c(-123, -75),  ylim = c(26, 41.8)) +
  scale_color_manual(values = c("dodgerblue2", "goldenrod"))+
  xlab("") + ylab("") +
  theme(panel.grid.minor=element_blank(),
        panel.grid.major = element_blank(),
        legend.position = "top",
        legend.title = element_text(size=9),
        panel.background=element_rect(color="white",fill="white",linewidth = 0.5),
        axis.line=element_line(color="black"),
        axis.text=element_text(color="black"),
        axis.ticks=element_line(color="black")) 



ggplot() +
  geom_polygon(data = world_map, aes(x=long, y = lat, group = group), 
               fill = "white", color="gray40", linewidth=0.3) + 
  geom_polygon(data = usa, aes(x=long, y = lat, group = group), 
               fill = "white", color="gray70", linewidth=0.3) + 
  geom_point(data=pop, aes(x=long, y = lat,color=Deme, size=depth),shape=19,alpha=0.7) +
  #geom_text(data=subset(pop,Deme=="pure"), aes(x=long, y = lat,label=depth),size=3.5) +
  geom_point(aes(x=-72.20392,y=44.14799),shape=23,size=2,stroke=2,fill="mediumvioletred",color="mediumvioletred")+
  coord_fixed(xlim = c(-123, -70),  ylim = c(26, 45)) +
  scale_color_manual(values = c("gray60", "black"))+
  xlab("") + ylab("") +
  theme(panel.grid.minor=element_blank(),
        panel.grid.major = element_blank(),
        legend.position = "top",
        panel.background=element_rect(color="white",fill="white",linewidth = 0.5),
        axis.line=element_blank(),
        axis.text=element_blank(),
        axis.ticks=element_blank()) 




ggplot() +
  geom_polygon(data = world_map, aes(x=long, y = lat, group = group), 
               fill = "gray95", color="gray40", linewidth=0.3) + 
  geom_polygon(data = usa, aes(x=long, y = lat, group = group), 
               fill = "gray95", color="gray70", linewidth=0.3) + 
  geom_point(data=pop, aes(x=long, y = lat,color=Subset, shape=Region), size=3) +
  coord_fixed(xlim = c(-123, -75),  ylim = c(26, 41.8)) +
  scale_color_manual(values = c("slateblue", "darkgoldenrod"))+
  xlab("") + ylab("") +
  theme(panel.grid.minor=element_blank(),
        panel.grid.major = element_blank(),
        legend.position = "top",
        legend.title = element_blank(),
        panel.background=element_rect(color="white",fill="white",linewidth = 0.5),
        axis.line=element_line(color="black"),
        axis.text=element_text(color="black"),
        axis.ticks=element_line(color="black")) 





ggplot() +
  #geom_polygon(data = world_map, aes(x=long, y = lat, group = group), 
              # fill = "white", color="gray40", linewidth=0.3) + 
  geom_polygon(data = usa, aes(x=long, y = lat, group = group), 
               fill = "white", color="gray70", linewidth=0.3) + 
  geom_point(data=pop, aes(x=long, y = lat,fill=Region),size=2, stroke=0.2,shape=21) +
  coord_fixed(xlim = c(-123, -75),  ylim = c(26, 41.8)) +
  scale_fill_manual(values = c("goldenrod2", "aquamarine3"))+
  #scale_color_manual(values = c("red", "black"))+
  xlab("") + ylab("") +
  theme(panel.grid.minor=element_blank(),
        panel.grid.major = element_blank(),
        legend.position = "none",
        panel.background=element_rect(color="white",fill="white",linewidth = 0.5),
        axis.line=element_blank(),
        axis.text=element_blank(),
        axis.ticks=element_blank()) 


ggsave("~/Downloads/foxmap12-eastwest_2.png", width=6, height = 2, units="in")


ggsave("~/Downloads/foxmap.png", width=6, height = 2, units="in")


```

### Genic Regions

create bed file of genes ±1kb from annotation

<details>
<summary>Show code</summary>
<br>
```{r annotation, eval=FALSE, warning=FALSE,message=FALSE}

#Grayfox liftover annotation

galba <- fread("~/Downloads/galba.gtf.gz")

cfanno <- fread("~/Downloads/canFam4.ncbiRefSeq.gtf.gz")

cfch <- read_tsv("~/Downloads/cf4_chroms.txt",col_names = F) 



cf3anno <- fread("~/Downloads/canfam3.1.refseq.gtf") %>%
  rename("V1"=`#!annotation-source NCBI Canis lupus familiaris Annotation Release 105`)

cf3ch <- read_tsv("~/Downloads/canfam3.1.report.tsv")  %>% filter(grepl("^chr[0-9]", `UCSC style name`)) %>% 
  select(`RefSeq seq accession`,`UCSC style name`,`Seq length`)  %>% 
  rename("scaf"=`RefSeq seq accession`,"chrom"=`UCSC style name`,"length"=`Seq length`)

cf3anf <- cf3anno %>% 
  filter(str_detect(V3, "CDS")) %>% mutate(len=V5-V4) %>% separate(V9, sep=";", into=c("gene","transcript")) %>% left_join(cf3ch,by=c("V1"="scaf")) 

cf3anch <- cf3anf  %>% group_by(gene) %>% slice(which.max(len)) %>% drop_na(chrom)

cf31kb <- cf3anch %>%  ungroup() %>% 
  mutate(newstart=V4-1000, newend=V5+1000) %>% 
  mutate(start=case_when(newstart<0 ~ 1, TRUE ~ newstart)) %>%
  mutate(end=case_when(newend>as.numeric(length) ~ as.numeric(length), TRUE ~ newend)) %>% dplyr::select(chrom,start,end)  %>% arrange(chrom,start)


cf31kb %>% write_tsv("sumstats_files/canfam3.1.gtfgenic.bed", col_names = F)




cfanf <- cfanno %>% 
  filter(str_detect(V3, "CDS")) %>% mutate(len=V5-V4) %>% separate(V9, sep=";", into=c("gene","transcript")) %>% left_join(cfch,by=c("V1"="X1")) 

cfanch <- cfanf  %>% drop_na(X2) %>% group_by(gene) %>% slice(which.max(len)) 

cf1kb <- cfanch %>%  ungroup() %>% 
  mutate(newstart=V4-1000, newend=V5+1000) %>% select(V1,newstart,newend) %>% arrange(V1,newstart)

cf1kb %>% write_tsv("sumstats_files/canfam4.gtfgenic.bed", col_names = F)


chroms <- read_tsv("inputstats_files/grayfox_renameChroms_number.txt", col_names = c("scaf","chrom")) %>% 
  separate(scaf, remove=F, c(NA,NA,NA,NA,NA,"length"))

gfgenes <- galba %>% left_join(chroms,by=c("V1"="scaf"))

mybednew <- gfgenes %>% 
  filter(str_detect(V3, "CDS")) %>% mutate(len=V5-V4) %>% separate(V9, sep=";", into=c("t","g"))

mybeds <- mybednew %>%  mutate(g = str_remove_all(g, "\"")) %>% separate(g, sep="d", into=c("id","gid")) %>% mutate(gid=as.numeric(str_remove(gid,"g"))) %>% arrange(gid,len)

mb <- mybeds %>% group_by(gid) %>% filter(len==max(len)) %>% drop_na(chrom) %>% ungroup()

mb %>% write_tsv("sumstats_files/gfgenes_2024.bed", col_names = F)

mb1kb <- mb %>% mutate(newstart=V4-1000, newend=V5+1000) %>% 
  mutate(start=case_when(newstart<0 ~ 1, TRUE ~ newstart)) %>%
  mutate(end=case_when(newend>as.numeric(length) ~ as.numeric(length), TRUE ~ newend)) %>%
  dplyr::select(chrom,start,end)

mb1kb %>% write_tsv("sumstats_files/gfgenes_2024_1kb.bed", col_names = F)

neutral <- read_tsv("sumstats_files/neutral.bed", col_names = c("chrom","start","end")) %>% mutate(size=end-start) 

neucut <- neutral %>% filter(size>1000) %>% 
  mutate(cutsize=cut(size,c(1e3,5e3,1e4,5e4,1e5,5e5,1e6,2e6,Inf),
                     labels=c('1kb-5kb','5kb-10kb','10kb-50kb','50kb-100kb','100kb-500kb','500kb-1mb','1mb-2mb','>2mb')))%>%
  mutate(cutsize=as.character(cutsize)) 

neutab <- neucut %>% group_by(cutsize) %>% count() 

ggplot(neutab,aes(x=cutsize, y=freq)) +
  geom_col() +
  theme_classic() + 
  theme(axis.text = element_text(color="black",size=10),
        axis.title = element_blank())

ggsave("~/Downloads/neutraldist.png", width=7.7, height = 4, units="in")



ggsave("sumstats_files/neutraldist.png", width=8, height = 3.5, units="in")

neuchrom <- neutral %>% filter(size>1000) %>% group_by(chrom) %>% summarize(length=sum(size)) %>% 
  mutate(chr=as.numeric(str_remove(chrom, "chr"))) %>% arrange(chr)

neureg <- neutral %>% filter(size>1000) %>% dplyr::select(size)





ggplot(neutab)+geom_histogram(aes(x=size))





neureg %>% write_tsv("sumstats_files/neutral_regions_gt1kb.txt", col_names = F)

neutral %>% filter(size>1000)  %>% dplyr::select(chrom,start,end) %>%  write_tsv("sumstats_files/neutral_regions_gt1kb.bed", col_names = F)

ggplot(neutral,aes(x=chrom,y=size)) + geom_boxplot()

neutralsmall <- neutral %>% filter(size<1000) %>% dplyr::select(chrom,start,end)

mbgt1kb <- bind_rows(mb1kb,neutralsmall) %>% arrange(chrom,start)

mbgt1kb %>% write_tsv("sumstats_files/gfgenes_2024_1kb_gt1kb.bed", col_names = F)

```
</details>


```{bash geneGC}
bedtools makewindows -g chrom_size.txt -w 100000 | bedtools sort -i - > temp.CalculateGDWindows.intervals.bed

bedtools makewindows -g chroms.bed -w 10000 | bedtools sort -i - > temp.CalculateGDWindows.intervals.10kb.bed

cp /project/jazlynmo_738/DataRepository/Reference_Files/GrayFox/liftoff_canfam3_grayfox_polished.gff ./
  
grep -v "^#" liftoff_canfam3_grayfox_polished.gff | awk '($3 == "gene") {print $1"\t"$4"\t"$5}' | bedtools sort -i - | bedtools merge -i - > temp.CalculateGDWindows.genes.bed

grep -v "^#" liftoff_canfam3_grayfox_polished.gff | awk '($3 == "exon") {print $1"\t"$4"\t"$5}' | bedtools sort -i - | bedtools merge -i - > temp.CalculateGDWindows.exons.bed

bedtools intersect -a temp.CalculateGDWindows.intervals.bed -b temp.CalculateGDWindows.genes.bed -wao | \
awk -v window_size=100000 '{count[$1"\t"$2"\t"$3]+=$7}END {for ( i in count) print i"\t"count[i] / window_size}' | \
bedtools sort -i - | sed -e 's/,/./g' | awk '{print $1"\t"$2"\t"$3"\t"$4}' - > gene_densities.txt

bedtools intersect -a temp.CalculateGDWindows.intervals.bed -b temp.CalculateGDWindows.exons.bed -wao | \
awk -v window_size=100000 '{count[$1"\t"$2"\t"$3]+=$7}END {for ( i in count) print i"\t"count[i] / window_size}' | \
bedtools sort -i - | sed -e 's/,/./g' | awk '{print $1"\t"$2"\t"$3"\t"$4}' - > exon_densities.txt

bedtools makewindows -g chrom_size.txt -w 10000 | bedtools nuc -fi /project/jazlynmo_738/DataRepository/Reference_Files/GrayFox/greyfox-dovetail-final.fasta -bed - | cut -f1-3,5 | tail -n +2 - > gc10kb.txt
```


```{bash emp_scratch}
bedtools intersect -a neutral_regions_gt1kb.bed -b /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr59_DPgr205lt500.vcf.gz > neutralSNPcount.txt

bcftools query -f '%CHROM %POS\n' /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr59_DPgr205lt500.vcf.gz  > snpquery.txt

module load gcc/11.3.0  
module load openblas/0.3.20

module load bcftools

bcftools view -Oz -S west6.txt /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr59_DPgr205lt500.vcf.gz > west6.vcf.gz

bcftools view -Oz -S east6.txt /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr59_DPgr205lt500.vcf.gz > east6.vcf.gz

bcftools index -t west6.vcf.gz

bcftools index -t east6.vcf.gz

bcftools view --min-ac=1 --max-ac 11 west6.vcf.gz | bcftools view -i 'F_MISSING<0.1' -Oz -o snpwest6.vcf.gz

bcftools index -t snpwest6.vcf.gz

bcftools view --min-ac=1 --max-ac 11 east6.vcf.gz | bcftools view -i 'F_MISSING<0.1' -Oz -o snpeast6.vcf.gz

bcftools index -t snpeast6.vcf.gz

bcftools query -f '%CHROM %POS\n' snpeast6.vcf.gz  > snpeast.txt

awk 'OFS="\t"{print $1, $2-1, $2}' snpeast.txt > snpeast.bed

vcftools --gzvcf snpeast6.vcf.gz --counts --out eastcount_neutral --exclude-bed gfgenes_2024_1kb_gt1kb.bed >& eastcount_neutral.log

vcftools --gzvcf snpwest6.vcf.gz --counts --out westcount_neutral --exclude-bed gfgenes_2024_1kb_gt1kb.bed >& westcount_neutral.log

sed -e 's/:/\t/g'  eastcount_neutral.frq.count > east_neutral.count

sed -e 's/:/\t/g'  westcount_neutral.frq.count > west_neutral.count

vcftools --gzvcf snpeast6.vcf.gz --counts --out eastcount >& eastcount.log

vcftools --gzvcf snpwest6.vcf.gz --counts --out westcount >& westcount.log

sed -e 's/:/\t/g'  eastcount.frq.count > east.count
sed -e 's/:/\t/g'  westcount.frq.count > west.count

vcftools --gzvcf snpwest6.vcf.gz --counts --out westcount --exclude-bed gfgenes_2024_1kb_gt1kb.bed >& westcount.log

bcftools query -f '%CHROM %POS\n' snpwest6.vcf.gz  > snpwest.txt

awk 'OFS="\t"{print $1, $2-1, $2}' snpwest.txt > snpwest.bed

module load bedtools2

bedtools intersect -a neutral_regions_gt1kb.bed -b snpeast.bed -c > neutral_SNP_east.txt

bedtools intersect -a neutral_regions_gt1kb.bed -b snpwest.bed -c > neutral_SNP_west.txt

awk 'OFS="\t"{print $1, $2-1, $2, $3}' east_sites.pi > eastpi.bed

awk 'OFS="\t"{print $1, $2-1, $2, $3}' east_sites_nozero.pi > eastpi.bed

awk 'OFS="\t"{print $1, $2-1, $2, $3}' allChroms.sites.pi > allChroms.sites.bed

bedtools intersect -abam chr${SLURM_ARRAY_TASK_ID} -b neutral_regions_gt1kb.bed -wb  > chr${SLURM_ARRAY_TASK_ID}_neutral_pi_east.txt

awk ' { print FILENAME","$0} ' cut*.csv > cut_all.csv

sed -e 's/cut_output_batch_//g' > cut_all.csv 

awk ' { print FILENAME"\t"$0} ' *.rmap > gf_east_n14.rmap

awk ' { print FILENAME"\t"$0} ' gf_east*.tsv > gf_east_n14_r2.tsv

awk ' { print FILENAME"\t"$0} ' gf_west*.tsv > gf_west_n12_r2.tsv

awk ' { print FILENAME"\t"$0} ' *_sumpi > allsumpi.txt

awk '{
    arr[$6]+=$4
   }
   END {
     for (key in arr) printf("%s\t%s\n", key, arr[key])
   }' chr${SLURM_ARRAY_TASK_ID}_neutral_pi_east.txt \
   | sort -k1,1 >chr${SLURM_ARRAY_TASK_ID}_sumpi

segeast <- read_tsv("~/Downloads/neutral_SNP_east.txt", col_names=c("chrom","start","end","snpcount")) %>% mutate(size=end-start)

pieast <- read_tsv("~/Downloads/allsumpi.txt", col_names=c("chrom","start","sumpi")) %>% 
  mutate(chrom=str_remove(chrom, "_sumpi"))
  
piseg <- segeast %>% left_join(pieast, by=c("chrom","start")) %>% mutate(avgpi=sumpi/size)


rpiseg <- replace(piseg,is.na(piseg), 0) 

tpiseg <- rpiseg %>% 
mutate(cutpi=as.numeric(as.character(cut(avgpi, seq(from = 0, to = 0.01, by = 0.0002), include.lowest = T, labels = c(0:49))))) 





rpisegcut <- rpiseg %>% 
mutate(cutpi=as.numeric(as.character(cut(avgpi, seq(from = 0, to = 0.01, by = 0.0002), include.lowest = T, labels = c(0:49))))) %>%
mutate(cutsnp=as.numeric(as.character(cut(snpcount, seq(from = 0, to = 36300, by = 726), include.lowest = T, labels = c(0:49)))))

mr <- seq(from=0,to=49,by=1) %>% as_data_frame() %>% rename("bin"=value)

rpi <- rpisegcut %>% group_by(cutpi) %>% count(name = "emp") %>% rename("bin"=cutpi) %>% right_join(mr)
rseg <- rpisegcut %>% group_by(cutsnp) %>% count(name = "emp") %>% rename("bin"=cutsnp)%>% right_join(mr)
 
rpi <- replace(rpi,is.na(rpi), 0) 
rseg <- replace(rseg,is.na(rseg), 0) 


ggplot(rpisegcut)+ geom_histogram(aes(snpcount),bins=100)+ theme_classic2()

ggplot(rpiseg)+geom_point(aes(x=propsnp,y=avgpi))

ggplot(rpiseg)+ geom_histogram(aes(snpcount),bins=50)+ theme_classic2()

ggplot(subset(rpiseg,snpcount>0))+ geom_histogram(aes(snpcount),bins=50)+ theme_classic2()

```


```{R recomb_talk}


sumstats<-fread("~/Downloads/empirical_bin_summary.txt")

ggplot(sumstats)+
  geom_col(aes(x=Bin,y=frequency,fill=pop))+
  facet_wrap(~summary+pop,scales="free")+
  #facet_grid(cols=vars(pop),rows=vars(summary),scales = "free")+ 
  scale_fill_manual(values = c("goldenrod2","aquamarine3")) +
  theme_classic()+
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    strip.background = element_blank(),
    axis.text = element_text(colour = "black", size = 10),
    axis.line = element_line(colour = "black", linewidth = 0.5),
    axis.ticks = element_line(colour = "black", linewidth = 0.5),
    axis.title.x = element_blank(),
    legend.position = "none")+  
  scale_y_continuous(labels = ~ format(.x, scientific = FALSE))



library(data.table)
library(tidyverse)
reast  <- fread("/scratch1/marjanak/mainland_2024/gf_correct/perChrom_recombination/gf_east_n14.rmap")
rwest  <- fread("/scratch1/marjanak/mainland_2024/gf_correct/perChrom_recombination/gf_west_n12.rmap")

rmapeast <-reast %>% 
mutate(V1=str_remove(V1, "gf_east_n14_")) %>%
mutate(V1=as.numeric(str_remove(V1, ".rmap")))
  
names(rmapeast) <- c("chrom","pos1", "pos2", "recomb_est")

window_size <- 20000

rmap_df <- rmapeast %>% 
      mutate(window_pos1 = floor(pos1 / window_size) * window_size) %>%
      mutate(window_pos2 = floor(pos1 / window_size) * 
               window_size + window_size) %>%
      group_by(chrom, window_pos1) %>%
      summarise(recomb_est_avg = mean(recomb_est, na.rm = TRUE)) %>%
      ungroup

ggplot(data=subset(rmap_df,chrom==1)) +
geom_point(aes(x=window_pos1,y=recomb_est_avg)) +
facet_wrap(~chrom, scales="free", nrow=4)  


ggplot(data=rmap_df) +
geom_point(aes(x=window_pos1,y=recomb_est_avg)) +
facet_wrap(~chrom, scales="free", nrow=4)  


write_tsv(rmap_df,"/scratch1/marjanak/mainland_2024/gf_correct/perChrom_recombination/east_rmap_5kb.txt")
 
rmap <- read_tsv("~/Downloads/rmap_10kb.txt") %>% mutate(pop=factor(pop,levels=c("west","east")))
  

rmap100 <- read_tsv("~/Downloads/rmap_100kb.txt") %>% mutate(pop=factor(pop,levels=c("west","east")))

rsum <- rmap %>% 
  group_by(pop) %>% 
  summarise(rmean=mean(recomb_est_avg),rmed=median(recomb_est_avg),
            rmin=min(recomb_est_avg),rmax=max(recomb_est_avg))

ggplot(rmap) + 
  geom_density(aes(x=log(recomb_est_avg),fill=pop),alpha=0.6,color=NA) +
  scale_fill_manual(values=c("blue","goldenrod2"))+
  theme_classic()+
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text.y = element_text(colour = "black", size = 10),
    axis.text.x = element_blank(),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", linewidth = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 12),
    legend.position = c(0.25,0.8),  
    legend.title = element_blank(),
    axis.title.y = element_text(colour = "black", size = 12))+
  xlab("log avg recombination rate")

ggsave("~/Downloads/recomb_densi.png", width=5, height = 2.5, units="in")



ggplot(rmap, aes(x = pop, y = log(recomb_est_avg), color=pop)) +
  geom_point(
    size = 1,
    alpha = .1,
    position = position_jitter(
      seed = 1, width = .2
    )
  ) + theme_classic()


ggplot(rmap) + 
  geom_density(aes(x=recomb_est_avg,color=pop)) +
  geom_vline(data=rsum, aes(xintercept=rmed, color=pop), linetype="dashed")+
  geom_vline(data=rsum, aes(xintercept=rmean, color=pop)) + facet_wrap(~pop,nrow=2)

rmap %>% group_by(pop) %>% summarise(mean(recomb_est_avg),median(recomb_est_avg))
  
  mutate(window_pos1=window_pos1/1e6) %>% unite("pos", chrom:window_pos1, sep=".", remove=T) 

ggplot(rmap,aes(x = pop, y = pos, fill = recomb_est_avg)) +
  geom_tile() 

rmapeast <- rmap %>% filter(pop=="east") %>% dplyr::select(chrom,window_pos1,recomb_est_avg)
rmapwest <- rmap %>% filter(pop=="west")%>% dplyr::select(chrom,window_pos1,recomb_est_avg)

rmapjoin <- rmapeast %>% inner_join(rmapwest,by=c("chrom","window_pos1"), suffix=c(".east",".west")) %>% mutate(window_pos1=window_pos1/1e6) %>% unite("pos", chrom:window_pos1, sep=".", remove=T)

rtest <- rmapjoin %>% distinct(pos, .keep_all=T) %>% t() %>% row_to_names(row_number=1) 
 
ggplot(rmapjoin,aes(x=log(recomb_est_avg.east),y=log(recomb_est_avg.west)))+
  geom_point()+
  geom_smooth(method = "lm", formula = y ~ x) 

#bedtools makewindows -g chroms.bed -w 10000 | bedtools sort -i - > temp.CalculateGDWindows.intervals.10kb.bed

#bedtools intersect -a temp.CalculateGDWindows.intervals.10kb.bed -b tss.bed -c > windowtss.bed


tsstest <- windowtss %>% left_join(rmapjoin,by=c("chrom","start"="window_pos1")) %>% drop_na() %>%  mutate(recomb=(recomb_est_avg.east+recomb_est_avg.west)/2)


 
ggplot() + 
  geom_point(data=tsstest, size=0.7,aes(x=start/1e6, y=recomb_est_avg.east*1e8, color=as.character(count))) +
  xlab("Position (mb)")+ylab("Recombination Rate (cM/Mb) in East")+
  scale_color_manual(values=c("gray80","#70b8fa","#5ea0e8","#4d89d5","#3e72c2","#315caf","#25469b","#1a3086","#0f1b71","#04035c"))+
  facet_grid(cols = vars(chrom),
             space = "free_x",  switch="x",
             scales = "free_x") +    
  labs(colour="TSS Count")+
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text.y = element_text(colour = "black", size = 10),
    axis.text.x = element_blank(),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", linewidth = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 12),
    legend.position="top",
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 12),   
    panel.spacing = unit(0.0, "cm"),
    strip.background=element_rect(fill="gray90"))+ 
  guides(colour = guide_legend(nrow = 1))

ggsave("~/Downloads/windowtss.png", width=11, height = 3.5, units="in")



epi<- read_tsv("~/Downloads/piwin_east.txt")
wpi<- read_tsv("~/Downloads/piwin_west.txt")

bothpi <- bind_rows("east"=epi,"west"=wpi,.id = "pop") %>% 
  mutate(chr.win=as.numeric(str_remove(chr.win, "chr")))

recombpi <- ewpi %>% mutate(BIN_START=BIN_START-1) %>% left_join(rmap, by=c("pop","CHROM"="chrom","BIN_START"="window_pos1"))


ggplot(recombpi,aes(x=recomb_est_avg*1e8,y=PI))+ geom_point() + 
  facet_wrap(~pop,nrow=2)+
  geom_smooth(method = "lm", formula = y ~ x, fullrange=T) + stat_cor(label.y = 0.018)+ 
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text = element_text(colour = "black", size = 12),
    axis.line = element_line(colour = "black", linewidth = 0.5),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 14),
    legend.position="top",
    legend.title = element_blank(),
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 14),   
    panel.spacing = unit(0.5, "cm"),
    strip.background=element_rect(fill="gray90"),  
    plot.margin = margin(10, 10, 10, 10))  + 
  xlab("Population Recombination Rate") +     
  ylab(expression(pi))

ggsave("~/Downloads/rrvspiS.png", width=4, height = 8, units="in")



recombfst <- fst10kb %>% mutate(BIN_START=BIN_START-1) %>%  left_join(rmapjoin,by=c("CHROM"="chrom","BIN_START"="window_pos1"))



ggplot(recombfst,aes(x=recomb_est_avg.east*1e8,y=MEAN_FST))+ geom_point() + 
    geom_smooth(method = "lm", formula = y ~ x) +
    stat_cor(label.y = 0.6,label.x = 6)+ theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text = element_text(colour = "black", size = 12),
    axis.line = element_line(colour = "black", linewidth = 0.5),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 14),
    legend.position="top",
    legend.title = element_blank(),
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 14),   
    panel.spacing = unit(0.5, "cm"),
    strip.background=element_rect(fill="gray90"),  
    plot.margin = margin(10, 10, 10, 10))  + 
  xlab("Population Recombination Rate") +     
  ylab("FST")+labs(title="East")

ggsave("~/Downloads/rrvsfst_east.png", width=4.5, height = 4, units="in")


fst10kb <- read_tsv("~/Downloads/all_east_west.windowed.weir.fst")%>% 
  mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) %>% filter(N_VARIANTS>5)




ggplot(fst10kb, aes(x = BIN_START/1e6, y=MEAN_FST, color=as.factor(CHROM))) +
scale_color_manual(values = rep(c("gray60", "black"), 100)) +
geom_point(size=0.5) + theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text = element_text(colour = "black", size = 12),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 14),
    legend.position="none",
    legend.title = element_blank(),
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 14),   
    panel.spacing = unit(0, "cm"),
    strip.background=element_rect(fill="gray90"),
    axis.text.x = element_blank())  +
  facet_grid(cols = vars(CHROM),
             space = "free_x", switch = "x",
             scales = "free_x") + 
  xlab("Position (Mb)") + 
  ylab(expression(paste(F["ST"]~" E vs. W")))

ggsave("~/Downloads/fstnew.png", width=11, height = 2.5, units="in")


e14pi <- read_tsv("~/Downloads/east_pi10kb.windowed.pi")
w12pi <- read_tsv("~/Downloads/west_pi10kb.windowed.pi")

ewpi <- bind_rows("west"=w12pi,"east"=e14pi, .id="pop")  %>% 
  mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) %>% mutate(pop=factor(pop,levels=c("west","east")))

ggplot(data=ewpi, aes(x=BIN_START/1e6, y=PI,color=as.factor(CHROM))) +
    scale_color_manual(values = rep(c("gray60", "black"), 100)) +
geom_point(size=0.5) + theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text = element_text(colour = "black", size = 12),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 14),
    legend.position="none",
    legend.title = element_blank(),
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 14),   
    panel.spacing.x  = unit(0, "cm"),
    panel.spacing.y  = unit(0.1, "cm"),
    strip.background=element_rect(fill="gray90"),
    axis.text.x = element_blank()) +
  facet_grid(cols = vars(CHROM), rows=vars(pop), switch="x",
             space = "free_x",
             scales = "free_x") +   
  xlab("Position (mb)")+
  ylab(expression(pi))

ggsave("~/Downloads/ew_pi.png", width=11, height = 4, units="in")


ggplot(data=ewpi, aes(x=BIN_START/1e6, y=PI,color=as.factor(CHROM))) +
    scale_color_manual(values = rep(c("gray60", "black"), 100)) +
#geom_point(size=0.5) + 
geom_smooth(method="loess",span=0.1, se=F, size=0.7) +
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text = element_text(colour = "black", size = 12),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 14),
    legend.position="none",
    legend.title = element_blank(),
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 14),   
    panel.spacing.x  = unit(0, "cm"),
    panel.spacing.y  = unit(0.1, "cm"),
    strip.background=element_rect(fill="gray90"),
    axis.text.x = element_blank()) +
  facet_grid(cols = vars(CHROM), rows=vars(pop), switch="x",
             space = "free_x",
             scales = "free_x") +   
  xlab("Position (mb)")+
  ylab(expression(pi))

ggsave("~/Downloads/ew_pi_smooth.png", width=11, height = 4, units="in")



rnames <- rmapjoin[,1]

names(df)<-make.names(names(df))



mat_data <- data.matrix(rmapjoin[,2:ncol(rmapjoin)])

ggplot(rmapjoin, aes(x = recomb_est_avg.east*1e8, y=recomb_est_avg.west*1e8)) +
geom_point()  +
  #geom_smooth(method="lm", size=0.7) +
    theme( panel.grid.minor=element_blank(),
      panel.grid.major=element_blank(),
      panel.background=element_blank(),
      axis.text = element_text(colour = "black", size = 8),
      axis.line = element_line(colour = "black", linewidth = 0.5),
      axis.ticks = element_line(colour = "black", linewidth = 0.5),
      axis.title = element_text(colour = "black", size = 10)) + 
    xlab("East Avg Recomb 10kbwin") + ylab("West  Avg Recomb 10kbwin") + facet_wrap(~chrom,nrow=4, scales="free")
    
ggsave("~/Downloads/e_w_rr_chroms.png", width=12, height = 6, units="in")

ggsave("~/Downloads/e_w_rr.png", width=5, height = 4, units="in")



ggplot(data=rmap, aes(x=window_pos1/1e6, y=recomb_est_avg*1e8,color=as.factor(chrom))) +
    scale_color_manual(values = rep(c("gray60", "black"), 100)) +
geom_point(size=0.5) + theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text = element_text(colour = "black", size = 12),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 14),
    legend.position="none",
    legend.title = element_blank(),
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 14),   
    panel.spacing = unit(0, "cm"),
    strip.background=element_rect(fill="gray90"),
    axis.text.x = element_blank()) +
  facet_grid(cols = vars(chrom), rows=vars(pop), switch="x",
             space = "free_x",
             scales = "free_x") +   
  xlab("Position (mb)")+
  ylab("Recombination Rate (cM/Mb)")

ggsave("~/Downloads/ew_rmap.png", width=11, height = 4, units="in")





ggplot(rmap,aes(x=pop,y=recomb_est_avg*1e8)) + 
  geom_boxplot() + stat_compare_means() + 
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text = element_text(colour = "black", size = 12),
    axis.line = element_line(colour = "black", linewidth = 0.5),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title = element_text(colour = "black", size = 14),
    axis.title.x = element_blank()) + ylab("Recombination Rate (cM/Mb)")

ggsave("~/Downloads/rrmeans.png", width=3.5, height =3.5 , units="in")



ggplot(data=rmap, aes(x=window_pos1/1e6, y=recomb_est_avg*1e8, color=as.factor(chrom))) + 
    scale_color_manual(values = rep(c("gray60", "black"), 100)) +
  geom_smooth(method="loess",span=0.1, se=F, size=0.7) +
  xlab("Position (mb)")+ylab("Recombination Rate (cM/Mb)")+
  facet_grid(cols = vars(chrom), rows=vars(pop),
             space = "free_x",  switch="x",
             scales = "free_x") +     
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text.y = element_text(colour = "black", size = 10),
    axis.text.x = element_blank(),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", linewidth = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 12),
    legend.position="none",
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 12),   
    panel.spacing = unit(0.0, "cm"),
    strip.background=element_rect(fill="gray90")) 

  
ggsave("~/Downloads/ew_rmap2.png", width=11, height = 4, units="in")
  

ggplot(rmap,aes(x=pop,y=recomb_est_avg)) + geom_boxplot()

reast  <- read_tsv("/scratch1/marjanak/mainland_2024/gf_correct/perChrom_recombination/gf_east_n14.rmap",col_names=c("chrom","start","end","rmap")) %>% 
mutate(chrom=str_remove(chrom, "gf_east_n14_")) %>%
mutate(chrom=as.numeric(str_remove(chrom, ".rmap")))


```


```{r talk-plots}

geneden <- read_tsv("~/Downloads/exon_densities.txt", col_names = F) %>% 
  left_join(chroms,by=c("X1"="scaf")) %>% 
  mutate(midpoint=(X2+X3)/2) %>% 
  mutate(chr=as.numeric(str_remove(chrom, "chr")))



gccon <- read_tsv("~/Downloads/gc100kb.txt", col_names = F) %>% 
  left_join(chroms,by=c("X1"="scaf")) %>% 
  mutate(midpoint=(X2+X3)/2) %>% 
  mutate(chr=as.numeric(str_remove(chrom, "chr")))


gccon10 <- read_tsv("~/Downloads/gc10kb.txt", col_names = F) %>% 
  left_join(chroms,by=c("X1"="scaf")) %>% 
  mutate(midpoint=(X2+X3)/2) %>% 
  mutate(chr=as.numeric(str_remove(chrom, "chr"))) %>% 
  dplyr::select(chr,X2,X4) 

names(gccon10) <- c("chrom","window_pos1", "gc")


  ggplot(geneden)+
  geom_point(aes(x=midpoint, y=X4), shape=20)+
  facet_grid(cols = vars(chr), space = "free", scales = "free") + 
  xlab("") + ylab("Exon density in 100kb windows")+
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    legend.position = "top",
    panel.spacing.x = unit(0.05, "cm"),
    axis.text = element_text(colour = "black", size = 12),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", linewidth = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 14),
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 14),   
    strip.background=element_rect(fill="gray90"),
    axis.text.x = element_blank()) 
  
  
  
  ggsave("~/Downloads/exonden.png", width=11, height = 3, units="in")
  
  
rgc <- rmap %>% left_join(gccon10, by=c("chrom","window_pos1"))
  
ggplot(rgc,aes(x=gc,y=recomb_est_avg*1e8))+
  geom_point() +
  geom_smooth(method = "lm", formula = y ~ x) +
  stat_cor(label.y = 5)+  
  facet_wrap(~pop)+
    theme( panel.grid.minor=element_blank(),
      panel.grid.major=element_blank(),
      panel.background=element_blank(),
      axis.text = element_text(colour = "black", size = 12),
      axis.line = element_line(colour = "black", linewidth = 0.5),
      axis.ticks = element_line(colour = "black", linewidth = 0.5),
      axis.title = element_text(colour = "black", size = 14)) + 
    xlab("GC Content") + ylab("Recombination Rate (cM/Mb)")

  
  
  ggplot(rgc, aes(x = gc, y=log(recomb_est_avg*1e8))) +
    geom_point() +
    geom_smooth(method = "lm", formula = y ~ x) +
    stat_cor(label.y = 5)+ 
    facet_wrap(~pop,nrow=2) +
    theme( panel.grid.minor=element_blank(),
      panel.grid.major=element_blank(),
      panel.background=element_blank(),
      axis.text = element_text(colour = "black", size = 12),
      axis.line = element_line(colour = "black", linewidth = 0.5),
      axis.ticks = element_line(colour = "black", linewidth = 0.5),
      axis.title = element_text(colour = "black", size = 14)) + 
    xlab("GC Content") + ylab("log avg recomb rate (cm/mb)")
  
    ggsave("~/Downloads/gccon3.png", width=4, height = 8, units="in")

  ggplot(gccon10,aes(x=window_pos1, y=gc))+
    #geom_point()+
    geom_smooth(color="black",method="loess",span=0.1, se=F, size=0.7) +
    facet_grid(cols = vars(chrom), space = "free", scales = "free") + 
    xlab("") + ylab("GC content 10kb-win ")+
    theme(
      panel.grid.minor=element_blank(),
      panel.grid.major=element_blank(),
      panel.background=element_blank(),
      legend.position = "top",
      panel.spacing.x = unit(0.05, "cm"),
      axis.text = element_text(colour = "black", size = 12),
      axis.line.y = element_line(colour = "black", linewidth = 0.5),
      axis.line.x = element_blank(),
      axis.ticks.x = element_blank(),
      axis.ticks.y = element_line(colour = "black", linewidth = 0.5),
      axis.ticks.length = unit(0.2, "cm"),
      axis.title.x = element_text(colour = "black", size = 14),
      strip.text = element_text(colour = "black", size = 12),
      axis.title.y = element_text(colour = "black", size = 14),   
      strip.background=element_rect(fill="gray90"),
      axis.text.x = element_blank()) 
  
  
  ggsave("~/Downloads/gccon2.png", width=11, height = 2.5, units="in")
  
    ggsave("~/Downloads/gccons.png", width=11, height = 2.5, units="in")


  
ggplot() + 
  geom_point(data=gccon10, aes(x=window_pos1/1e6, y=gc, color=as.factor(chrom))) +
  scale_color_manual(values = rep(c("gray60", "black"), 100)) +
  xlab("Position (mb)")+ylab("GC Content")+
  facet_grid(cols = vars(chrom),
             space = "free_x",  switch="x",
             scales = "free_x") +     
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text.y = element_text(colour = "black", size = 10),
    axis.text.x = element_blank(),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", linewidth = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 12),
    legend.position="none",
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 12),   
    panel.spacing = unit(0.0, "cm"),
    strip.background=element_rect(fill="gray90")) 
  
  
```



### Summary Statistics 


### Split VCF by pop and count alleles 
<details>
<summary>Show code </summary>
<br>
```{bash sepvcf, eval=FALSE}
#!/bin/sh
#SBATCH --job-name=sepvcf
#SBATCH --output=/scratch1/marjanak/sepvcf.out
#SBATCH --error=/scratch1/marjanak/sepvcf.err
#SBATCH --partition=qcb
#SBATCH --time=10:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem-per-cpu=8000MB
#SBATCH --mail-type=END,FAIL # notifications for job done & fail
#SBATCH --mail-user=marjanak@usc.edu

module load bcftools

bcftools view -Oz -S west6.txt /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.drop295.ACgr25_DPgr165lt500.vcf.gz > west6.vcf.gz

bcftools view -Oz -S east6.txt /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.drop295.ACgr25_DPgr165lt500.vcf.gz > east6.vcf.gz

bcftools index -t west6.vcf.gz

bcftools index -t east6.vcf.gz

vcftools --gzvcf east6.vcf.gz --counts --out eastcount --max-missing 1.0 --exclude-bed gfgenes_2024.bed >& eastcount.log

vcftools --gzvcf west6.vcf.gz --counts --out westcount --max-missing 1.0 --exclude-bed gfgenes_2024.bed >& westcount.log

```
</details>

<br>


```{r gfcounts, warning=FALSE, eval=FALSE, message=FALSE, echo=FALSE}

eastcount <- fread("~/Downloads/east.frq.count",col.names = c("CHROM","POS","N_ALLELES","N_CHR","allele1","count1","allele2","count2"))


westcount <- fread("~/Downloads/west.frq.count",col.names = c("CHROM","POS","N_ALLELES","N_CHR","allele1","count1","allele2","count2"))


joincount <- full_join(eastcount,westcount,by=c("CHROM","POS"), suffix=c(".east",".west") )

eastct <- joincount %>% 
  mutate(east=case_when(is.na(N_CHR.east) ~ "missing",
                        count1.east==N_CHR.east ~ "fixed",
                        count2.east==N_CHR.east ~ "fixed",
                        TRUE ~ "polymorphic"))  %>% 
  mutate(west=case_when(is.na(N_CHR.west) ~ "missing",
                        count1.west==N_CHR.west ~ "fixed",
                        count2.west==N_CHR.west ~ "fixed",
                        TRUE ~ "polymorphic"))

tab <- eastct %>% group_by(east,west) %>% 
  dplyr::summarise(count=n())  %>% ungroup() %>% arrange(desc(count)) %>% 
  mutate(perc = percent(count / sum(count), accuracy = 0.1))


allcount <- fread("~/all.frq.count",col.names = c("CHROM","POS","N_ALLELES","N_CHR","allele1","count1","allele2","count2")) 


alc <- allcount %>% 
  mutate(snp=case_when( count1==N_CHR ~ "fixed anc",
                        count2==N_CHR ~ "fixed derived",
                        TRUE ~ "polymorphic"))  

tgf <- alc %>% group_by(snp)  %>% 
  dplyr::summarise(n_gf=n())  %>% ungroup() %>% arrange(desc(n_gf)) %>% 
  mutate(perc_gf = percent(n_gf / sum(n_gf), accuracy = 0.1))


cfcount <- fread("~/Downloads/cfall.frq.count",col.names = c("CHROM","POS","N_ALLELES","N_CHR","allele1","count1","allele2","count2")) 


calc <- cfcount %>% 
  mutate(snp=case_when( count1==N_CHR ~ "fixed anc",
                        count2==N_CHR ~ "fixed derived",
                        TRUE ~ "polymorphic"))  

tcf <- calc %>% group_by(snp)  %>% 
  dplyr::summarise(n_cf=n())  %>% ungroup() %>% arrange(desc(n_cf)) %>% 
  mutate(perc_cf = percent(n_cf / sum(n_cf), accuracy = 0.1))


newtab <- tgf %>% left_join(tcf) %>% adorn_totals() %>%
  mutate(n_gf=comma(n_gf),n_cf=comma(n_cf))

## canfam3.1

cfeastcount <- fread("~/Downloads/eastcf.frcq.count",col.names = c("CHROM","POS","N_ALLELES","N_CHR","allele1","count1","allele2","count2"))

cfwestcount <- fread("~/Downloads/westcf.frq.count",col.names = c("CHROM","POS","N_ALLELES","N_CHR","allele1","count1","allele2","count2"))



cfec <- cfeastcount %>% group_by(count1,count2) %>% count() 
cfwc <- cfwestcount %>% group_by(count1,count2) %>% count() 

cfc <- left_join(cfec,cfwc,by=c("count1","count2"),suffix=c(".east",".west") )

bothcc <- left_join(gfc,cfc,by=c("count1","count2"),suffix=c(".gf",".cf") ) %>%
  mutate(p.east.gf=percent(n.east.gf/499097,accuracy = 0.1)) %>%
  mutate(p.west.gf=percent(n.west.gf/524003,accuracy = 0.1)) %>% 
  mutate(p.east.cf=percent(n.east.cf/19461006,accuracy = 0.1)) %>%
  mutate(p.west.cf=percent(n.west.cf/19876196,accuracy = 0.1)) %>%   
  adorn_totals() %>% select(count1,count2,n.east.gf,p.east.gf,n.west.gf,p.west.gf,n.east.cf,p.east.cf,n.west.cf,p.west.cf)



cfjoincount <- full_join(cfeastcount,cfwestcount,by=c("CHROM","POS"), suffix=c(".east",".west") )

cfeastct <- cfjoincount %>% 
  mutate(east=case_when(is.na(N_CHR.east) ~ "missing",
                        count1.east==N_CHR.east ~ "fixed",
                        count2.east==N_CHR.east ~ "fixed",
                        TRUE ~ "polymorphic"))  %>% 
  mutate(west=case_when(is.na(N_CHR.west) ~ "missing",
                        count1.west==N_CHR.west ~ "fixed",
                        count2.west==N_CHR.west ~ "fixed",
                        TRUE ~ "polymorphic"))

cftab <- cfeastct %>% group_by(east,west) %>% 
  dplyr::summarise(count=n())  %>% ungroup() %>% arrange(desc(count)) %>% 
  mutate(perc = percent(count / sum(count), accuracy = 0.1))


jointab <- inner_join(tab,cftab,by=c("east","west"), suffix=c(".gf",".cf") ) %>% adorn_totals() 

write_tsv(jointab,"sumstats_files/jointab.tsv")
```


```{r tablecount, warning=FALSE, eval=TRUE, message=FALSE, echo=FALSE}

jointab <- read_tsv("sumstats_files/jointab.tsv")


```


<br>




```{bash sitepi}
#!/bin/sh
#SBATCH --job-name=sitepi
#SBATCH --output=/scratch1/marjanak/slurmJob_%a.out
#SBATCH --error=/scratch1/marjanak/slurmJob_%a.err
#SBATCH --time=30:00:00
#SBATCH --partition=qcb
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem-per-cpu=8000MB
#SBATCH --array=1-38
#SBATCH --mail-type=END,FAIL # notifications for job done & fail
#SBATCH --mail-user=marjanak@usc.edu

module purge
module load vcftools

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Invariant/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr59_DPgr205lt500.gvcf.gz --keep east6.txt --site-pi --chr ${SLURM_ARRAY_TASK_ID} --max-missing 1.0 --out /scratch1/marjanak/pi_east/chr${SLURM_ARRAY_TASK_ID}sitepi_East

sleep 60


awk 'FNR>1' chr* > west_sites.pi
awk 'NR > 1 {sum+=$3} END {print sum / (NR - 1)}' west_sites.pi 


awk 'FNR>1' chr* > east_sites.pi
awk 'NR > 1 {sum+=$3} END {print sum / (NR - 1)}' east_sites.pi 

west <- scientific(2.32456e-05,digits = 3)

2.32e-05

1.11e-04


east <- scientific(0.000111182,digits = 3)


vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.drop295.ACgr25_DPgr165lt500.vcf.gz --exclude-bed gfgenes_2024_1kb.bed --keep empirical.txt --het --out grayfox.het
 
vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.drop295.ACgr25_DPgr165lt500.vcf.gz --exclude-bed gfgenes_2024_1kb.bed --keep empirical.txt --het --freqx --out gf_het_freq



vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/Canfam3.1_filtered.renameChroms.Mainland.drop295.ACgr59_DPgr205lt500.vcf.gz --keep empirical.txt --het --out canfam.het





```


```{r smcplot, eval=F}
plotboth <- read_csv("~/plot_gf.csv") 



plotboth%>%group_by(label,y) %>% rename("Ne"="y") %>% summarise(end=max(x),start=min(x)) %>%  arrange(label,desc(start)) 

ploteastG <- read_csv("~/Downloads/plot_gf_east_feb15.csv") %>% left_join()
plotwestG <- read_csv("~/Downloads/plot_gf_west_feb15.csv") 


cx_breaks = c(1e3, 1e4, 1e5, 1e6, 5e6)
cy_breaks = c(0,25e2,5e3,75e2,125e2,25e3,5e4,75e3,15e4,2e5,5e5)

gfplotwest <- read_csv("~/Downloads/plot_gf_west_n12.csv")
gfploteast <- read_csv("~/Downloads/plot_gf_east_n14.csv")

gfpb<- bind_rows("east"=gfploteast,"west"=gfplotwest,.id = "pop") 

cfploteast <- read_csv("~/Downloads/plot_east_canfam4.csv")
cfplotwest <- read_csv("~/Downloads/plot_west_canfam4.csv")

cfpb <- bind_rows("east"=cfploteast,"west"=cfplotwest,.id = "pop")  %>% mutate(x=x/2)

cfgf <- bind_rows("gf"=gfpb, "cf4"=cfpb, .id="genome") %>% filter(x>1e3 & x<1e6)

ggplot()+
  geom_line(data=cfgf,aes(x=x,y=y,color=pop,linetype=genome),linewidth=1) + 
  scale_y_continuous(trans = "log",breaks=cy_breaks)+  
  scale_x_continuous(trans = "log",breaks=cx_breaks)+ 
  labs(caption = "4.5e-9 mutation rate",
       x ="Generations", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11),legend.position = "top") +
  scale_color_manual(values = c("goldenrod2","aquamarine3")) 

ggsave("~/Downloads/alldem_cfgf.png", width=5, height = 4, units="in")



ploteast <- read_csv("~/plot_gf_east.csv") %>% 
  mutate(iteration=case_when(plot_num<1 ~ "final", 
                             TRUE ~ "iteration"))

plotwest <- read_csv("~/plot_gf_west.csv") %>% 
  mutate(iteration=case_when(plot_num<1 ~ "final", 
                             TRUE ~ "iteration"))

ploteastf <- ploteast %>% filter(iteration=="final")
plotwestf <- plotwest %>% filter(iteration=="final")


pb <- bind_rows("east"=ploteast,"west"=plotwest,.id = "pop") %>% filter(x>2e3)

zx_breaks = c(1e3, 1e4, 1e5, 1e6, 5e6)
zy_breaks = c(0,25e3,5e4,75e3,1e5,15e4,2e5,12e4)



ggplot()+
  geom_line(data=subset(pb,plot_num>0),aes(x=x,y=y,group=plot_num,color=plot_num)) + 
  scale_y_continuous(trans = "log",breaks=zy_breaks)+  
  scale_x_continuous(trans = "log",breaks=zx_breaks)+ 
  labs(caption = "4.5e-9 mutation rate",
       x ="Generations", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11),legend.position = "top") +
  scale_color_continuous(high = "#132B43", low = "#56B1F7") + 
  guides(color=guide_legend(title="EM iteration")) + facet_wrap(~pop, nrow=2) + 
  geom_vline(xintercept = 2e3, linetype="dotted", 
                color = "red", size=0.5)


ggsave("~/Downloads/alldem_new_it.png", width=4.3, height = 7, units="in")


pb%>%filter(iteration=="final") %>% group_by(label,y) %>% rename("Ne"="y") %>% summarise(end=max(x),start=min(x)) %>%  arrange(label,desc(start)) %>% mutate(rNe=round_any(Ne,1000))

pb%>%filter(iteration=="final") %>% group_by(label,y) %>% dplyr::rename("Ne"="y") %>% dplyr::summarise(end=max(x),start=min(x)) %>%  arrange(label,desc(start)) %>% mutate(Ne=round_any(Ne,1000),start=round_any(start,1000),end=round_any(end,1000))


plotW <- ggplot()+
  geom_line(data=subset(pb, iteration=="final" & x>2e3 & x<1e6 & pop=="west"),aes(x=x,y=y),color="aquamarine3",size=2) + 
  scale_y_continuous(trans = "log",breaks=zy_breaks)+  
  scale_x_continuous(trans = "log",breaks=zx_breaks,labels = ~ format(.x, scientific = FALSE))+ 
  labs(x ="Generations", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11),
        legend.title = element_blank(),legend.position = "bottom") +
  #scale_color_manual(values = c("goldenrod2","aquamarine3"))  + 
  #geom_vline(xintercept = 1e6, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=1.15e6, y=5e4, label="1M", angle=90,size=3)+
  #geom_vline(xintercept = 210e3, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=235e3, y=5e4, label="210k", angle=90,size=3)+
  geom_vline(xintercept = 135e3, linetype="dotted", color = "black", size=0.5)+ 
  annotate("text", x=175e3, y=5e4, label="135k",size=4)+
  #geom_vline(xintercept = 100e3, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=115e3, y=5e4, label="100k", angle=90,size=3)+
  #geom_vline(xintercept = 35e3, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=40e3, y=5e4, label="35k", angle=90,size=3)+
  #geom_vline(xintercept = 13.5e3, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=15e3, y=5e4, label="13.5k", angle=90,size=3)+
  geom_vline(xintercept = 3e3, linetype="dotted", color = "black", size=0.5)+
  annotate("text", x=3.5e3, y=5e4, label="3k",size=4)

ggsave("~/Downloads/westdemest.png", width=7.5, height = 4.5, units="in")


plotE <- ggplot()+
  geom_line(data=subset(pb, iteration=="final" & x>2e3 & x<1e6 & pop=="east"),aes(x=x,y=y),color="goldenrod2",size=2) + 
  scale_y_continuous(trans = "log",breaks=zy_breaks)+  
  scale_x_continuous(trans = "log",breaks=zx_breaks,labels = ~ format(.x, scientific = FALSE))+ 
  labs(x ="Generations", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11),
        legend.title = element_blank(),legend.position = "bottom") +
  #scale_color_manual(values = c("goldenrod2","aquamarine3"))  + 
  #geom_vline(xintercept = 1e6, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=1.15e6, y=5e4, label="1M", angle=90,size=3)+
  #geom_vline(xintercept = 210e3, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=235e3, y=5e4, label="210k", angle=90,size=3)+
  #geom_vline(xintercept = 135e3, linetype="dotted", color = "black", size=0.5)+ 
  #annotate("text", x=155e3, y=5e4, label="135k", angle=90,size=3)+
  geom_vline(xintercept = 112e3, linetype="dotted", color = "black", size=0.5)+
  annotate("text", x=13e4, y=5e4, label="112k",size=4)+
  #geom_vline(xintercept = 35e3, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=40e3, y=5e4, label="35k", size=4)+
  #geom_vline(xintercept = 62e3, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=67e3, y=5e4, label="62k", angle=90,size=3)+
  #geom_vline(xintercept = 14e3, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=16e3, y=5e4, label="14k", angle=90,size=3)+
  geom_vline(xintercept = 3e3, linetype="dotted", color = "black", size=0.5)+
  annotate("text", x=3.3e3, y=5e4, label="3k",,size=4)


ggsave("~/Downloads/eastdemest.png", width=7.5, height = 4.5, units="in")


grid.arrange(plotW,plotE,nrow=2)
ggsave("~/Downloads/alldem_epo3.png", width=5, height = 10, units="in")

ggplot()+
  geom_line(data=subset(pb, iteration=="final" & x>900 & x<1e6),aes(x=x,y=y,color=pop),size=2,alpha=0.8) + 
  scale_y_continuous(trans = "log",breaks=zy_breaks)+  
  scale_x_continuous(trans = "log",breaks=zx_breaks)+ 
  labs(x ="Generations", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11),
        legend.title = element_blank(),legend.position = "bottom") +
  scale_color_manual(values = c("goldenrod2","aquamarine3"))  + 
  #geom_vline(xintercept = 1e6, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=1.15e6, y=3e4, label="1,000,000", angle=90,size=3)+
  geom_vline(xintercept = 210e3, linetype="dotted", color = "black", size=0.5)+
  annotate("text", x=235e3, y=3e4, label="210,000", angle=90,size=3)+
  #geom_vline(xintercept = 135e3, linetype="dotted", color = "aquamarine4", size=0.5)+ 
 # annotate("text", x=155e3, y=5e4, label="135,000", angle=90,size=3,color="aquamarine4")+
  geom_vline(xintercept = 100e3, linetype="dotted", color = "black", size=0.5)+
  annotate("text", x=115e3, y=3e4, label="100,000", angle=90,size=3)+
  geom_vline(xintercept = 35e3, linetype="dotted", color = "black", size=0.5)+
  annotate("text", x=40e3, y=5e4, label="35,000", angle=90,size=3)+
  geom_vline(xintercept = 3.3e3, linetype="dotted", color = "black", size=0.5)+
  annotate("text", x=3.8e3, y=5e4, label="3,300", angle=90,size=3)+
  #geom_vline(xintercept = 2e3, linetype="dotted", color = "black", size=0.5)+
  #annotate("text", x=2.3e3, y=5e4, label="2,000", angle=90,size=3)+
  theme(plot.margin = margin(2,0.5,0.5,0.5, "cm"))

ggsave("~/Downloads/alldem_epoxxx.png", width=7, height = 5, units="in")



ggplot()+
  geom_line(data=subset(pb,iteration=="final" & x>2e3),aes(x=x,y=y,color=pop)) + 
  scale_y_continuous(trans = "log",breaks=zy_breaks)+  
  scale_x_continuous(trans = "log",breaks=zx_breaks)+ 
  labs(caption = "4.5e-9 mutation rate",
       x ="Generations", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11),legend.position = "top") +
  scale_color_manual(values = c("goldenrod2","aquamarine3")) 

ggsave("~/Downloads/alldem_new.png", width=6, height = 5, units="in")


ggplot()+
  geom_line(data=subset(pb,iteration=="iteration"),aes(x=x,y=y,group=plot_num, color=plot_num)) + 
  geom_line(data=subset(pb,iteration=="model"),aes(x=x,y=y,group=plot_num), linetype="dotted") + 
  scale_y_continuous(trans = "log",breaks=zy_breaks)+ 
  scale_x_continuous(trans = "log",breaks=zx_breaks)+
  labs(caption = "4.5e-9 mutation rate",
       x ="Generations", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11),legend.position = "top")  +
  scale_color_continuous(high = "#132B43", low = "#56B1F7") +
  facet_wrap(~pop)  + guides(color=guide_legend(title="EM iteration"))





ploteast <- read_csv("sumstats_files/plot_gf_east.csv") %>% filter(x>1e3 & x<1e7)

plotwest <- read_csv("sumstats_files/plot_gf_west.csv") %>% filter(x>1e3 & x<1e7)


ploteastall <- read_csv("sumstats_files/plot_gf_east_all2.csv") %>% filter(x>8e2 & x<1e6)

plotwestall <- read_csv("sumstats_files/plot_gf_west_all2.csv") %>% filter(x>8e2 & x<1e6)

allplot <- bind_rows("east"=ploteastall, "west"=plotwestall, .id="pop")

ax_breaks = c(0,10,1e2, 1e3, 1e4, 1e5, 1e6, 1e7)
ay_breaks = c(25,50,1e2,250,500,1e3,25e2,6e3,15e3,3e4)

ggplot()+
  geom_line(data=allplot,aes(x=x,y=y,color=pop)) + 
  scale_y_continuous(trans = "log",breaks=ay_breaks)+ 
  scale_x_continuous(trans = "log",breaks=ax_breaks)+
  labs(caption = "4.5e-9 mutation rate and 2 year generation",
       x ="Years", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11),legend.position = "top") +
  scale_color_manual(values = c("goldenrod2","aquamarine3"))

ggsave("sumstats_files/alldem.png", width=6, height = 5, units="in")


ex_breaks = c(0,1,10,1e2, 1e3, 1e4, 1e5, 1e6, 1e7)
ey_breaks = c(1e3,25e2,6e3,15e3,25e3,5e4,1e5)



ggplot()+
  geom_line(data=plotwestall,aes(x=x,y=y,group=plot_num), color="aquamarine3") + 
  scale_y_continuous(trans = "log",breaks=ey_breaks)+ 
  scale_x_continuous(trans = "log",breaks=ex_breaks)+
  labs(title="western gray fox demography per chromosome",
       caption = "4.5e-9 mutation rate and 2 year generation",
       x ="Years", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11)) 



ploteastit <- read_csv("sumstats_files/plot_gf_east_all_it.csv")  %>% filter(x>8e2 & x<1e6) %>% filter(plot_num>2)

plotwestit <- read_csv("sumstats_files/plot_gf_west_all_it.csv")  %>% filter(x>8e2 & x<1e6) %>% filter(plot_num>4)



ox_breaks = c(1e3, 1e4, 1e5, 1e6)
oy_breaks = c(5e2,15e2,4e3,1e4,3e4,6e4)

ggplot()+
  geom_line(data=ploteastit,aes(x=x,y=y,group=plot_num), color="goldenrod2",alpha=0.5) + 
  geom_line(data=plotwestit,aes(x=x,y=y,group=plot_num), color="aquamarine3",alpha=0.5) + 
  geom_line(data=ploteastall,aes(x=x,y=y,group=plot_num), color="goldenrod4") + 
  geom_line(data=plotwestall,aes(x=x,y=y,group=plot_num), color="aquamarine4") + 
  scale_y_continuous(trans = "log",breaks=oy_breaks)+ 
  scale_x_continuous(trans = "log",breaks=ox_breaks)+
  labs(title="gray fox demography",
       caption = "4.5e-9 mutation rate and 2 year generation",
       x ="Years", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11)) 


ggplot()+
  geom_line(data=ploteastit,aes(x=x,y=y,group=plot_num), color="goldenrod2",alpha=0.5) + 
  geom_line(data=plotwestit,aes(x=x,y=y,group=plot_num), color="aquamarine3",alpha=0.5) + 
  geom_line(data=ploteastall,aes(x=x,y=y,group=plot_num), color="goldenrod4") + 
  geom_line(data=plotwestall,aes(x=x,y=y,group=plot_num), color="aquamarine4") + 
  scale_y_continuous(trans = "log",breaks=oy_breaks)+ 
  scale_x_continuous(trans = "log",breaks=ox_breaks)+
  labs(title="gray fox demography",
       caption = "4.5e-9 mutation rate and 2 year generation",
       x ="Years", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11)) 

ggsave("sumstats_files/alldem2.png", width=6, height = 5, units="in")


plotwestall%>%group_by(y) %>% rename("Ne"="y") %>% summarise(end=max(x),start=min(x)) %>% arrange(desc(start))

ploteastall%>%group_by(y) %>% rename("Ne"="y") %>% summarise(end=max(x),start=min(x)) %>% arrange(desc(start))




ggplot()+
  geom_line(data=ploteast,aes(x=x,y=y,group=plot_num), color="goldenrod2") + 
  scale_y_continuous(trans = "log",breaks=ey_breaks)+ 
  scale_x_continuous(trans = "log",breaks=ex_breaks)+
  labs(title="eastern gray fox demography per chromosome",
       caption = "4.5e-9 mutation rate and 2 year generation",
       x ="Years", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11)) 

ggsave("sumstats_files/eastdem.png", width=6, height = 5, units="in")


x_breaks = c(0,1,10,1e2, 1e3, 1e4, 1e5, 1e6, 1e7)
y_breaks = c(25,50,1e2,250,500,1e3,25e2,6e3,15e3,3e4)

ggplot()+
  geom_line(data=plotwest,aes(x=x,y=y,group=plot_num), color="aquamarine3") + 
  scale_y_continuous(trans = "log",breaks=y_breaks)+ 
  scale_x_continuous(trans = "log",breaks=x_breaks)+
  labs(title="western gray fox demography per chromosome",
       caption = "4.5e-9 mutation rate and 2 year generation",
       x ="Years", y = "Effective Population Size") +
  theme_classic() +
  theme(axis.text=element_text(color="black", size=11)) 
ggsave("sumstats_files/westdem.png", width=6, height = 5, units="in")
```


```{r fst, eval=F}

fst <- read_tsv("~/Downloads/all_east_west.windowed.weir.fst") %>%
  mutate(chr_position = ((BIN_START + BIN_END)/2)/1000000) %>%
  mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) %>% filter(N_VARIANTS>10)



ggplot() + 
  geom_point(data=fst, aes(x=chr_position, y=WEIGHTED_FST), size=0.15) +
  facet_wrap(~CHROM,nrow = 1,scales = "free_x")+
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text = element_text(colour = "black", size = 12),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 14),
    legend.position="top",
    legend.title = element_blank(),
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 14),   
    panel.spacing = unit(0.0, "cm"),
    strip.background=element_rect(fill="gray90"),
    axis.text.x = element_blank()) 

```




```{r pca, eval=F}




mypop <- read_tsv("inputstats_files/mainland_WGSmetadata.txt") %>% 
  mutate(group=case_when(Deme=="hybrid" ~"hybrid", TRUE ~ Region)) %>% 
  filter(Dataset=="Sacks") %>% mutate(State=factor(State, levels = c("CA","NV","NM","TX","OK","AR","MO","MS","TN","AL","KY","SC"))) 

myfiltpop <- mypop %>% filter(!group=="hybrid") %>% 
  dplyr::select(Region,SRA,Sex) %>% arrange(Region,Sex)

mypca <- read_tsv("~/pcagfnewest.tsv") %>% left_join(mypop,by=c("sample.id"="SRA")) 


cfpca  <- read_tsv("~/Downloads/pcacf.tsv") %>% left_join(mypop,by=c("sample.id"="SRA"))
4.63 2.80 2.65 2.64 2.63 2.62

ggplot(mypca,aes(x=EV2,y=EV1, label=stateID))+
  geom_text(size=2,position = position_jitter(w = 0.1, h = 0.1))+
  theme_classic() +
  labs(title="Grayfox vcf") + 
  theme(axis.text = element_text(color="black",size=12),
        axis.title = element_text(size=14))

ggplot(mypca)+
  geom_point(aes(x=EV1,y=EV2,color=State),size=1,position = position_jitter(w = 0, h = 0.04)) +
  theme_classic() +
  labs(x ="PC1 (4.63 PVE)", y = "PC2 (2.80 PVE)") + 
  scale_color_manual(values=c("#F01FFF","#8B1FFF","#261FFF","#1F7CFF","#1FE1FF","#1FFFB8","#1FFF53","#4FFF1F","#B4FF1F","#FFE51F","#FF801F","#FF1F1F"))+
  theme(axis.text = element_text(color="black",size=12),
        axis.title = element_text(size=14)) + scale_x_reverse() + scale_y_reverse()

ggsave("~/Downloads/grayfox_vcf_pca_new.png", width=6, height = 5, units="in")


ggplot(mypca)+
  geom_point(aes(x=EV1,y=EV2,color=State), shape=19,size=2) +
  scale_color_brewer(palette = "Paired") +
  labs(x ="PC1", y = "PC2") + theme_classic() + 
  theme(axis.text = element_text(color="black",size=12),
        axis.title = element_text(size=14)) + scale_x_reverse()

ggsave("~/grayfox_vcf_pca.png", width=6, height = 5, units="in")

  
ggplot(mypca)+
  geom_point(aes(x=EV2,y=EV3,color=State), shape=19,size=2) +
  scale_color_brewer(palette = "Paired") +
  labs(x ="PC2 (7.66 PVE)", y = "PC3 (5.84 PVE)") + theme_classic() + 
  theme(axis.text = element_text(color="black",size=12),
        axis.title = element_text(size=14))

ggsave("~/grayfox_vcf_pca23.png", width=6, height = 5, units="in")


  
ggplot(cfpca)+
  geom_point(aes(x=EV1,y=EV1,color=State,shape=group),size=1,position = position_jitter(w = 0, h = 0.03)) +
  theme_classic() +
  labs(title="Canfam3.1 vcf", x ="PC1 (11.36%)", y = "PC1 jitter") + 
  scale_color_manual(values=c("#F01FFF","#8B1FFF","#261FFF","#1F7CFF","#1FE1FF","#1FFFB8","#1FFF53","#4FFF1F","#B4FF1F","#FFE51F","#FF801F","#FF1F1F"))+
  theme(axis.text = element_text(color="black",size=12),
        axis.title = element_text(size=14)) + scale_x_reverse()

ggsave("~/cf_pc1.png", width=6, height = 5, units="in")

ggplot(mypca)+
  geom_point(aes(x=EV1,y=EV1,color=State,shape=group),size=1,position = position_jitter(w = 0, h = 0.03)) +
  theme_classic() +
  labs(title="Grayfox vcf", x ="PC1 (22.6%)", y = "PC1 jitter") + 
  scale_color_manual(values=c("#F01FFF","#8B1FFF","#261FFF","#1F7CFF","#1FE1FF","#1FFFB8","#1FFF53","#4FFF1F","#B4FF1F","#FFE51F","#FF801F","#FF1F1F"))+
  theme(axis.text = element_text(color="black",size=12),
        axis.title = element_text(size=14)) + scale_x_reverse()


ggsave("~/Downloads/gf_pc1.png", width=6, height = 5, units="in")





  
ggplot(mypca)+
  geom_point(aes(x=EV1,y=EV2,color=State,shape=group),size=1) +
  theme_classic() +
  scale_color_manual(values=c("#F01FFF","#8B1FFF","#261FFF","#1F7CFF","#1FE1FF","#1FFFB8","#1FFF53","#4FFF1F","#B4FF1F","#FFE51F","#FF801F","#FF1F1F"))+
  theme(axis.text = element_text(color="black",size=12),
        axis.title = element_text(size=14)) + scale_x_reverse()

ggplot(cfpca)+
  geom_point(aes(x=EV1,y=EV2,color=State,shape=group),size=1) +
  theme_classic() +
  scale_color_manual(values=c("#F01FFF","#8B1FFF","#261FFF","#1F7CFF","#1FE1FF","#1FFFB8","#1FFF53","#4FFF1F","#B4FF1F","#FFE51F","#FF801F","#FF1F1F"))+
  theme(axis.text = element_text(color="black",size=12),
        axis.title = element_text(size=14)) + scale_x_reverse()

ggplot(cfpca)+
  geom_point(aes(x=EV1,y=EV2,color=State,shape=group),size=1) +
  theme_classic() +
  labs(x ="PC1", y = "PC2") + 
  scale_color_manual(values=c("#F01FFF","#8B1FFF","#261FFF","#1F7CFF","#1FE1FF","#1FFFB8","#1FFF53","#4FFF1F","#B4FF1F","#FFE51F","#FF801F","#FF1F1F"))+
  theme(axis.text = element_text(color="black",size=12),
        axis.title = element_text(size=14)) + scale_x_reverse()




ggsave("~/grayfox_vcf_pca13.png", width=6, height = 5, units="in")




het <- read_tsv("sumstats_files/newhet.txt") %>%  
  mutate(samp=as.numeric(str_remove(INDV, "SRR24465"))) 


ggplot(het)+
  geom_boxplot(aes(x=genome,y=HET,color=pop))+
  scale_color_manual(values = c("#4d9aff", "#f2675f"))+
  ylab("heterozygosity") + theme_classic() + 
  theme(axis.text = element_text(color="black",size=12),
        axis.title = element_text(size=14))


ggplot(het)+
  geom_col(aes(x=pop, y=HET, group=samp, fill=genome), position = "dodge" )+
  scale_color_manual(values = c("#4d9aff", "#f2675f"))+
  scale_fill_manual(values = c("brown", "gray"))+
  ylab("heterozygosity") + theme_classic() + 
  theme(axis.text = element_text(color="black",size=12),
        axis.title = element_text(size=14))

ggplot(het)+
  geom_col(aes(x=INDV,y=HET,fill=genome, color=pop), position = "dodge" )+
  scale_color_manual(values = c("#4d9aff", "#f2675f"))+
  scale_fill_manual(values = c("brown", "gray"))+
  ylab("heterozygosity") + theme_classic() + 
  theme(axis.text = element_text(color="black",size=12),
  axis.title = element_text(size=14))
  
ggsave("sumstats_files/newhet.png", width=6, height = 5, units="in")



gfeastpi <- read_tsv("~/Downloads/gf_east_sites.pi") %>% filter(!CHROM=="CHROM") %>% 
  mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) 

gfwestpi <- read_tsv("~/Downloads/gf_west_sites.pi")  %>% filter(!CHROM=="CHROM") %>% 
  mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) 

gfallpi <- bind_rows("east"=gfeastpi,"west"=gfwestpi,.id = "pop") 

pitab <- gfallpi %>% group_by(pop) %>% summarise(avgpi=round(mean(PI),3), sdpi=round(sd(PI),3), variants=n())

ggplot(data = gfallpi, aes(x = as.numeric(PI))) +
  geom_histogram(bins=100) +
  facet_wrap(~pop)


cfeastpi <- fread("~/Downloads/cf_east_sites.pi")  %>% filter(!CHROM=="CHROM") %>% 
  mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) 

cfwestpi <- fread("~/Downloads/cf_west_sites.pi")  %>% filter(!CHROM=="CHROM") %>% 
  mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) 


cfallpi <- bind_rows("east"=cfeastpi,"west"=cfwestpi,.id = "pop") 

cfpitab <- cfallpi %>% group_by(pop) %>% summarise(avgpi=round(mean(PI),3), sdpi=round(sd(PI),3), variants=n())

newtab <- bind_rows("grayfox"=pitab,"canfam3.1"=cfpitab, .id="genome")
knitr::kable(newtab, format = "html")


cfeastpi %>% mutate(PI=as.numeric(PI)) %>% 
  mutate(rearranged=case_when(CHROM %in% c(3, 4, 5, 8, 10, 12, 14, 15, 16, 17, 20, 21, 23, 24, 27, 34) ~ "no", TRUE ~ "yes")) %>% 
  group_by(rearranged) %>% 
  summarise(avgpi=round(mean(PI),5), sdpi=round(sd(PI),4), variants=n())


cfwestpi %>% mutate(PI=as.numeric(PI)) %>% 
  mutate(rearranged=case_when(CHROM %in% c(3, 4, 5, 8, 10, 12, 14, 15, 16, 17, 20, 21, 23, 24, 27, 34) ~ "no", TRUE ~ "yes")) %>% 
  group_by(rearranged) %>% 
  summarise(avgpi=round(mean(PI),5), sdpi=round(sd(PI),4), variants=n())


ggplot() + 
  geom_line(data=gfallpi, aes(x=POS, y=PI, color=pop), size=0.15) +
  scale_alpha_manual(values = rep(c(1, 0.7), 100)) +
  scale_color_manual(values = c("#4d9aff", "#f2675f"))+
  xlab("Position (Mbp)") +
  facet_grid(cols = vars(CHROM), rows=vars(pop),
             space = "free_x",
             scales = "free_x") +     
  ylab(expression(pi))+
  #scale_y_continuous(limits = c(0,0.025), breaks = c(0, 0.01, 0.02))+
  theme(
    panel.grid.minor=element_blank(),
    panel.grid.major=element_blank(),
    panel.background=element_blank(),
    axis.text = element_text(colour = "black", size = 12),
    axis.line.y = element_line(colour = "black", linewidth = 0.5),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_line(colour = "black", size = 0.5),
    axis.ticks.length = unit(0.2, "cm"),
    axis.title.x = element_text(colour = "black", size = 14),
    legend.position="top",
    legend.title = element_blank(),
    strip.text = element_text(colour = "black", size = 12),
    axis.title.y = element_text(colour = "black", size = 14),   
    panel.spacing = unit(0.0, "cm"),
    strip.background=element_rect(fill="gray90"),
    axis.text.x = element_blank())  

 test1 <- cfeastct %>% filter(str_detect(east, "fixed") )  %>% 
   mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) %>% 
   mutate(rearranged=case_when(
     CHROM %in% c(3, 4, 5, 8, 10, 12, 14, 15, 16, 17, 20, 21, 23, 24, 27, 34) ~ "no", 
     TRUE ~ "yes")) %>% 
   group_by(east,west,rearranged) %>% count()
 
 test2 <- cfeastct %>% filter(str_detect(west, "fixed") ) %>% 
   mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) %>% 
   mutate(rearranged=case_when(
     CHROM %in% c(3, 4, 5, 8, 10, 12, 14, 15, 16, 17, 20, 21, 23, 24, 27, 34) ~ "no", 
     TRUE ~ "yes")) %>% 
   group_by(east,west,rearranged) %>% count()
 
 test3 <- left_join(test1,test2,by=c("CHROM"))
 
 
 test1 <- cfeastct %>% filter(east=="fixed" & west=="polymorphic") %>% group_by(east,west,CHROM) %>% count()
 test2 <- cfeastct %>% filter(west=="fixed" & east=="polymorphic") %>% group_by(east,west,CHROM) %>% count()

 test3 <- left_join(test1,test2,by=c("CHROM")) %>% 
   mutate(CHROM=as.numeric(str_remove(CHROM, "chr"))) %>% 
   mutate(rearranged=case_when(
     CHROM %in% c(3, 4, 5, 8, 10, 12, 14, 15, 16, 17, 20, 21, 23, 24, 27, 34) ~ "no", 
     TRUE ~ "yes"))
 
 bcftools view -Oz -S east6.txt /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr25_DPgr165lt500.vcf.gz
 
 bcftools query -f '%CHROM\t%POS\t%REF\t%ALT[\t%SAMPLE=%GT]\n' -r chr10:300000-350000 -S empirical.txt /project/jazlynmo_738/DataRepository/Canids/Invariant/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr59_DPgr205lt500.rmMultAllelic.gvcf.gz
 
 
 
 bcftools query -f '%CHROM\t%POS\t%FS\t%SOR\t%MQRankSum\t%ReadPosRankSum\t%QD\t%MQ\t%DP\n' -S empirical.txt /project/jazlynmo_738/DataRepository/Canids/Invariant/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr59_DPgr205lt500.rmMultAllelic.gvcf.gz > data.txt
 
vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Invariant/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr59_DPgr205lt500.rmMultAllelic.gvcf.gz --freq --keep empirical.txt --chr 10 --out chr10_analysis


 
scores <- read_tsv("~/Downloads/data.txt", col_names = c("CHR","POS","FS","SOR","MQRS","RPRS","QD","MQ", "DP")) %>% 
  mutate(CHR=as.numeric(str_remove(CHR, "chr"))) 
 
 
chr10 <- fread("~/Downloads/fscore.txt") 
 
 
 
 
emiss <- fread("~/Downloads/east.lmiss") %>% 
  mutate(CHR=as.numeric(str_remove(CHR, "chr"))) 


wmiss <- fread("~/Downloads/west.lmiss") %>% 
  mutate(CHR=as.numeric(str_remove(CHR, "chr"))) 


easttest <-gfeastpi %>% left_join(emiss,by=c("CHROM"="CHR","POS"))
westtest <-gfwestpi %>% left_join(wmiss,by=c("CHROM"="CHR","POS"))

ggplot() + 
  geom_point(data=easttest, aes(x=PI, N_MISS))




vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr59_DPgr205lt500.vcf.gz --keep empirical.txt --het --out grayfox.het
 



vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/Mainland/grayfox_filtered.renameChroms.Mainland.drop295.ACgr59_DPgr205lt500.vcf.gz --keep empirical.txt --het --out grayfox.het
 
vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/ChannelIsland/grayfox_filtered.renameChroms.ACgr25_DPgr165lt500.vcf.gz --het --max-missing 1.0 --out CI_grayfox.het

vcftools --gzvcf /project/jazlynmo_738/DataRepository/Canids/Variants/GrayFox/ChannelIsland/Canfam3.1_filtered.renameChroms.ACgr25_DPgr165lt500.vcf.gz --het --max-missing 1.0 --out CI_canfam31.het
```







```{bash smc-scratch}
#!/bin/sh
#SBATCH --job-name=smcpp_cf
#SBATCH --output=/scratch1/marjanak/smcpp_cf_%a.out
#SBATCH --error=/scratch1/marjanak/smcpp_cf_%a.err
#SBATCH --time=30:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem-per-cpu=8000MB
#SBATCH --array=1-38
#SBATCH --mail-type=END,FAIL # notifications for job done & fail
#SBATCH --mail-user=marjanak@usc.edu


module purge
module load gcc/12.3.0
module load singularity/3.11.3


singularity run -C --bind $PWD --pwd $PWD /project/jazlynmo_738/software/smcpp_latest.sif vcf2smc Canfam3.1_filtered.renameChroms.ACgr25_DPgr165lt500.vcf.gz chr${SLURM_ARRAY_TASK_ID}.cf.smc.gz chr${SLURM_ARRAY_TASK_ID} Pop1:GFO41F

singularity run -C --bind $PWD --pwd $PWD /project/jazlynmo_738/software/smcpp_latest.sif estimate 2e-8 --em-iterations 25 chr${SLURM_ARRAY_TASK_ID}*.cf.smc.gz -o cf --base chr${SLURM_ARRAY_TASK_ID}



singularity run -C --bind $PWD --pwd $PWD /project/jazlynmo_738/software/smcpp_latest.sif estimate 2e-8 --em-iterations 25 chr*.cf.smc.gz -o cf --base cf


#!/bin/sh
#SBATCH --job-name=smcpp_est
#SBATCH --output=/scratch1/marjanak/smcppest.out
#SBATCH --error=/scratch1/marjanak/smcppest.err
#SBATCH --time=30:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=4
#SBATCH --mem-per-cpu=8000MB
#SBATCH --mail-type=END,FAIL # notifications for job done & fail
#SBATCH --mail-user=marjanak@usc.edu


module purge
module load gcc/12.3.0
module load singularity/3.11.3


singularity run -C --bind $PWD --pwd $PWD /project/jazlynmo_738/software/smcpp_latest.sif estimate 2e-8 --em-iterations 25 --timepoints 500 1000000 --Nmin 0 --Nmax 1.6e4 --knots 10 chr*.cf.smc.gz



singularity run -C --bind $PWD --pwd $PWD /project/jazlynmo_738/software/smcpp_latest.sif plot plot_cf.png model.final.json 

singularity run -C --bind $PWD --pwd $PWD /project/jazlynmo_738/software/smcpp_latest.sif estimate 2e-8 --em-iterations 25 --timepoints 500 1000000 --Nmin 0 --Nmax 1.6e4 --knots 10 chr*.cf.smc.gz -o cf --base cf


```

